---
layout: post
title: options setting for blast
date: '2012-08-03T22:26:00.000-04:00'
author: Xianjun Dong
tags:
- blast
- bioinformatics
modified_time: '2012-08-03T22:26:33.750-04:00'
blogger_id: tag:blogger.com,1999:blog-12445049.post-5438188402755603926
blogger_orig_url: http://onetipperday.blogspot.com/2012/08/options-setting-for-blast.html
---

Blast is a classical method for finding homolog sequences. In one of my recent projects, I need to find the orthologous sequences of&nbsp;<i><a href="http://en.wikipedia.org/wiki/Manduca_sexta" target="_blank">Manduca sexta</a></i>&nbsp;(a moth) protein in other relative species. &nbsp;Since I want to get orthologs for all Manduca's proteins, which is a long list, a straightforward way is to do blast.<br /><br />First, you need to download/install blast. I use the ncbi-blast-2.2.26+-src.tar.gz downloaded from: ftp://ftp.ncbi.nlm.nih.gov/blast/executables/blast+/LATEST/<br /><br />Next, you need to download the blast database. Here is the list: ftp://ftp.ncbi.nlm.nih.gov/blast/db/. I used the&nbsp;nr.*tar.gz&nbsp;(non-redundant protein sequence database). If you want to use a smaller and specific one, you can choose for example&nbsp;pdbaa.*tar.gz.<br /><br />Once you installed the program and database, you can run the executable file called "blastall" in the bin folder. Here are its options:<br /><br /><br /><span style="font-family: Courier New, Courier, monospace;">blastall 2.2.26 &nbsp; arguments:</span><br /><span style="font-family: Courier New, Courier, monospace;"><br /></span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; -p &nbsp;Program Name [String]</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; -d &nbsp;Database [String]</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; &nbsp; default = nr</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; -i &nbsp;Query File [File In]</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; &nbsp; default = stdin</span><br /><b style="font-family: 'Courier New', Courier, monospace;">## alignment</b><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; -F &nbsp;Filter query sequence (DUST with blastn, SEG with others) [String]</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; &nbsp; default = T</span><br /><span style="font-family: 'Courier New', Courier, monospace;">&nbsp; -Q &nbsp;Query Genetic code to use [Integer]</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; &nbsp; default = 1</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; -D &nbsp;DB Genetic code (for tblast[nx] only) [Integer]</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; &nbsp; default = 1</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; -W &nbsp;Word size, default if zero (blastn 11, megablast 28, all others 3) [Integer]</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; &nbsp; default = 0</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; -z &nbsp;Effective length of the database (use zero for the real size) [Real]</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; &nbsp; default = 0</span><br /><span style="font-family: 'Courier New', Courier, monospace;">&nbsp; -Y &nbsp;Effective length of the search space (use zero for the real size) [Real]</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; &nbsp; default = 0</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; -S &nbsp;Query strands to search against database (for blast[nx], and tblastx)</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; &nbsp; &nbsp; &nbsp;3 is both, 1 is top, 2 is bottom [Integer]</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; &nbsp; default = 3</span><br /><span style="font-family: 'Courier New', Courier, monospace;">&nbsp; -M &nbsp;Matrix [String]</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; &nbsp; default = BLOSUM62</span><br /><span style="font-family: Courier New, Courier, monospace;"><b>## scoring&nbsp;</b></span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; -G &nbsp;Cost to open a gap (-1 invokes default behavior) [Integer]</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; &nbsp; default = -1</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; -E &nbsp;Cost to extend a gap (-1 invokes default behavior) [Integer]</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; &nbsp; default = -1</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; -X &nbsp;X dropoff value for gapped alignment (in bits) (zero invokes default behavior)</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; &nbsp; &nbsp; blastn 30, megablast 20, tblastx 0, all others 15 [Integer]</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; &nbsp; default = 0</span><br /><span style="font-family: 'Courier New', Courier, monospace;">&nbsp; -q &nbsp;Penalty for a nucleotide mismatch (blastn only) [Integer]</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; &nbsp; default = -3</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; -r &nbsp;Reward for a nucleotide match (blastn only) [Integer]</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; &nbsp; default = 1</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; -f &nbsp;Threshold for extending hits, default if zero</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; &nbsp; &nbsp; blastp 11, blastn 0, blastx 12, tblastn 13</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; &nbsp; &nbsp; tblastx 13, megablast 0 [Real]</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; &nbsp; default = 0</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; -g &nbsp;Perform gapped alignment (not available with tblastx) [T/F]</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; &nbsp; default = T</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; -y &nbsp;X dropoff value for ungapped extensions in bits (0.0 invokes default behavior)</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; &nbsp; &nbsp; blastn 20, megablast 10, all others 7 [Real]</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; &nbsp; default = 0.0</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; -Z &nbsp;X dropoff value for final gapped alignment in bits (0.0 invokes default behavior)</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; &nbsp; &nbsp; blastn/megablast 100, tblastx 0, all others 25 [Integer]</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; &nbsp; default = 0</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; -w &nbsp;Frame shift penalty (OOF algorithm for blastx) [Integer]</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; &nbsp; default = 0</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; -t &nbsp;Length of the largest intron allowed in a translated nucleotide sequence when linking multiple distinct alignments. (0 invokes default behavior; a negative value disables linking.) [Integer]</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; &nbsp; default = 0</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; -C &nbsp;Use composition-based score adjustments for blastp or tblastn:</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; &nbsp; &nbsp; As first character:</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; &nbsp; &nbsp; D or d: default (equivalent to T)</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; &nbsp; &nbsp; 0 or F or f: no composition-based statistics</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; &nbsp; &nbsp; 2 or T or t: Composition-based score adjustments as in Bioinformatics 21:902-911,</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; &nbsp; &nbsp; 1: Composition-based statistics as in NAR 29:2994-3005, 2001</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; 2005, conditioned on sequence properties</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; &nbsp; &nbsp; 3: Composition-based score adjustment as in Bioinformatics 21:902-911,</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; 2005, unconditionally</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; &nbsp; &nbsp; For programs other than tblastn, must either be absent or be D, F or 0.</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;As second character, if first character is equivalent to 1, 2, or 3:</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; &nbsp; &nbsp; U or u: unified p-value combining alignment p-value and compositional p-value in round 1 only</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp;[String]</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; &nbsp; default = D</span><br /><span style="font-family: Courier New, Courier, monospace;"><b>## display</b></span><br /><span style="color: red; font-family: Courier New, Courier, monospace;">&nbsp; -I &nbsp;Show GI's in deflines [T/F]</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; &nbsp; default = F</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; -v &nbsp;Number of database sequences to show one-line descriptions for (V) [Integer]</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; &nbsp; default = 500</span><br /><span style="color: red; font-family: Courier New, Courier, monospace;">&nbsp; -b &nbsp;Number of database sequence to show alignments for (B) [Integer]</span><br /><span style="color: red; font-family: Courier New, Courier, monospace;">&nbsp; &nbsp; default = 250</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; -K &nbsp;Number of best hits from a region to keep. Off by default.</span><br /><span style="font-family: Courier New, Courier, monospace;">If used a value of 100 is recommended. &nbsp;Very high values of -v or -b is also suggested [Integer]</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; &nbsp; default = 0</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; -P &nbsp;0 for multiple hit, 1 for single hit (does not apply to blastn) [Integer]</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; &nbsp; default = 0</span><br /><b style="font-family: 'Courier New', Courier, monospace;">## output</b><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; -O &nbsp;SeqAlign file [File Out] &nbsp;Optional</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; -J &nbsp;Believe the query defline [T/F]</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; &nbsp; default = F</span><br /><span style="font-family: 'Courier New', Courier, monospace;">&nbsp; -T &nbsp;Produce HTML output [T/F]</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; &nbsp; default = F</span><br /><span style="color: red; font-family: Courier New, Courier, monospace;">&nbsp; -e &nbsp;Expectation value (E) [Real]</span><br /><span style="color: red; font-family: Courier New, Courier, monospace;">&nbsp; &nbsp; default = 10.0</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; -m &nbsp;alignment view options:</span><br /><span style="font-family: Courier New, Courier, monospace;">0 = pairwise,</span><br /><span style="font-family: Courier New, Courier, monospace;">1 = query-anchored showing identities,</span><br /><span style="font-family: Courier New, Courier, monospace;">2 = query-anchored no identities,</span><br /><span style="font-family: Courier New, Courier, monospace;">3 = flat query-anchored, show identities,</span><br /><span style="font-family: Courier New, Courier, monospace;">4 = flat query-anchored, no identities,</span><br /><span style="font-family: Courier New, Courier, monospace;">5 = query-anchored no identities and blunt ends,</span><br /><span style="font-family: Courier New, Courier, monospace;">6 = flat query-anchored, no identities and blunt ends,</span><br /><span style="color: red; font-family: Courier New, Courier, monospace;">7 = XML Blast output,</span><br /><span style="font-family: Courier New, Courier, monospace;">8 = tabular,&nbsp;</span><br /><span style="font-family: Courier New, Courier, monospace;">9 tabular with comment lines</span><br /><span style="font-family: Courier New, Courier, monospace;">10 ASN, text</span><br /><span style="font-family: Courier New, Courier, monospace;">11 ASN, binary [Integer]</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; &nbsp; default = 0</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; &nbsp; range from 0 to 11</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; -o &nbsp;BLAST report Output File [File Out] &nbsp;Optional</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; &nbsp; default = stdout</span><br /><span style="font-family: 'Courier New', Courier, monospace;"><b>## performance</b></span><br /><span style="font-family: 'Courier New', Courier, monospace;"><span style="color: red;">&nbsp; -a &nbsp;Number of processors to use [Integer]</span></span><br /><span style="color: red; font-family: Courier New, Courier, monospace;">&nbsp; &nbsp; default = 1</span><br /><span style="font-family: 'Courier New', Courier, monospace;">&nbsp; -l &nbsp;Restrict search of database to list of GI's [String] &nbsp;Optional</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; -U &nbsp;Use lower case filtering of FASTA sequence [T/F] &nbsp;Optional</span><br /><span style="font-family: 'Courier New', Courier, monospace;">&nbsp; -R &nbsp;PSI-TBLASTN checkpoint file [File In] &nbsp;Optional</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; -n &nbsp;MegaBlast search [T/F]</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; &nbsp; default = F</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; -L &nbsp;Location on query sequence [String] &nbsp;Optional</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; -A &nbsp;Multiple Hits window size, default if zero (blastn/megablast 0, all others 40 [Integer]</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; &nbsp; default = 0</span><br /><span style="font-family: 'Courier New', Courier, monospace;">&nbsp; -B &nbsp;Number of concatenated queries, for blastn and tblastn [Integer] &nbsp;Optional</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; &nbsp; default = 0</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; -V &nbsp;Force use of the legacy BLAST engine [T/F] &nbsp;Optional</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; &nbsp; default = F</span><br /><span style="font-family: 'Courier New', Courier, monospace;">&nbsp; -s &nbsp;Compute locally optimal Smith-Waterman alignments (This option is only</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; &nbsp; &nbsp; available for gapped tblastn.) [T/F]</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; &nbsp; default = F</span><br /><br />I highlight the important ones with read.<br /><br /><b id="internal-source-marker_0.6720505771227181" style="font-weight: normal;"><span style="font-family: Arial; font-size: 15px; vertical-align: baseline; white-space: pre-wrap;">Here is the command I finally used to blast protein sequences:</span></b><br /><b style="font-weight: normal;"><span style="font-family: Arial; font-size: 15px; vertical-align: baseline; white-space: pre-wrap;"><br /></span></b><br /><b style="font-weight: normal;"><span style="font-family: Arial; font-size: 15px; vertical-align: baseline; white-space: pre-wrap;">blastall -p blastp -d ~/nearline/blast/nr/nr -i ~/nearline/genomes/Manduca_sexta/Jun2012/Msex05162011.genome.except-5small-scaf.maker-2.25.proteins.OGS_June2012.fasta -o Msex05162011.genome.except-5small-scaf.maker-2.25.proteins.OGS_June2012.fasta.blastp.top15.xml -e 1 -b 15 -m 7 -a 8</span></b><br />