---
layout: post
title: understand the FLAG code of SAM format
date: '2012-04-03T17:35:00.002-04:00'
author: Xianjun Dong
tags:
- next-generation sequencing
- FLAG
- SAM
modified_time: '2012-05-15T14:12:35.715-04:00'
blogger_id: tag:blogger.com,1999:blog-12445049.post-8579931042280227252
blogger_orig_url: http://onetipperday.blogspot.com/2012/04/understand-flag-code-of-sam-format.html
---

<br />Here is what SAM specification stated for the FLAG column:<br /><blockquote class="tr_bq"><span style="font-family: 'Courier New', Courier, monospace;">FLAG: bitwise FLAG. Each bit is explained in the following table:<br />Bit Description<br />0x1 template having multiple segments in sequencing (1)<br />0x2 each segment properly aligned according to the aligner (2)<br />0x4 segment unmapped (4)<br />0x8 next segment in the template unmapped (8)<br />0x10 SEQ being reverse complemented (16)<br />0x20 SEQ of the next segment in the template being reversed (32)<br />0x40 the rst segment in the template (64)<br />0x80 the last segment in the template (128)<br />0x100 secondary alignment (256)<br />0x200 not passing quality controls (512)<br />0x400 PCR or optical duplicate (1024)</span></blockquote>These codes are hexadecimal (also base 16, or hex)  code, e.g.&nbsp;0x10 = 16 in base ten format (decimal). It's easy to convert hex to binary format, simply convert each bit in hex format to 4 bits, e.g. 1=0001, 0=0000, so, 0x10=00010000 in binary way.<br /><br />In SAM format, the 2nd column (i.e. FLAG above) is one of the mandatory fields in the alignment section. It can be used to indicate the alignment, for example, 0x4 means unmapped reads, 0x10 means the reads mapped to the minus strand. &nbsp;In the SAM format, it uses decimal format, so imagine&nbsp;64+16+2+1=83 means it's first read (0x40) of pair-end reads (0x1) and it's mapped on minus strand (0x10) and both reads mapped (0x2).&nbsp;&nbsp;Here is a very nice post from Kamalakar giving the full&nbsp;explanation&nbsp;of the code:<br /><br /><a href="http://seqanswers.com/forums/showthread.php?t=17314">http://seqanswers.com/forums/showthread.php?t=17314</a><br /><br /><span style="background-color: #f5f5ff; font-family: verdana, geneva, lucida, 'lucida grande', arial, helvetica, sans-serif; font-size: 13px;">For paired reads, 0'th bit HAS to be set. Hence all flags for paired reads HAVE to be odd. In other words, all even-numbered flags other than the above three (0, 4 and 16) are meaningless&nbsp;</span><u style="font-family: verdana, geneva, lucida, 'lucida grande', arial, helvetica, sans-serif; font-size: 13px;"><span style="color: blue;">(Note: this may not totally true; for multiple hits, 256 is used for alignments beyond the first one. See below)</span></u><span style="background-color: #f5f5ff; font-family: verdana, geneva, lucida, 'lucida grande', arial, helvetica, sans-serif; font-size: 13px;">. (</span><a class="bigusername" href="http://seqanswers.com/forums/member.php?u=21909" style="background-color: #e1e4f2; color: #ff4400; font-family: verdana, geneva, lucida, 'lucida grande', arial, helvetica, sans-serif; font-size: 13px; font-weight: bold;">kgulukota</a><span style="background-color: #f5f5ff; font-family: verdana, geneva, lucida, 'lucida grande', arial, helvetica, sans-serif; font-size: 13px;">)</span><br /><br /><span style="background-color: #f5f5ff; font-family: verdana, geneva, lucida, 'lucida grande', arial, helvetica, sans-serif; font-size: 13px;">The four good numbers to remember are 64+16+2+1=</span><b style="background-color: #f5f5ff; font-family: verdana, geneva, lucida, 'lucida grande', arial, helvetica, sans-serif; font-size: 13px;">83</b><span style="background-color: #f5f5ff; font-family: verdana, geneva, lucida, 'lucida grande', arial, helvetica, sans-serif; font-size: 13px;">, 64+32+2+1=</span><b style="background-color: #f5f5ff; font-family: verdana, geneva, lucida, 'lucida grande', arial, helvetica, sans-serif; font-size: 13px;">99</b><span style="background-color: #f5f5ff; font-family: verdana, geneva, lucida, 'lucida grande', arial, helvetica, sans-serif; font-size: 13px;">, 128+16+2+1=</span><b style="background-color: #f5f5ff; font-family: verdana, geneva, lucida, 'lucida grande', arial, helvetica, sans-serif; font-size: 13px;">147</b><span style="background-color: #f5f5ff; font-family: verdana, geneva, lucida, 'lucida grande', arial, helvetica, sans-serif; font-size: 13px;">&nbsp;and 128+32+2+1=</span><b style="background-color: #f5f5ff; font-family: verdana, geneva, lucida, 'lucida grande', arial, helvetica, sans-serif; font-size: 13px;">163</b><span style="background-color: #f5f5ff; font-family: verdana, geneva, lucida, 'lucida grande', arial, helvetica, sans-serif; font-size: 13px;">. Something is very wrong if you ever see both 128 and 64 together, and with most current technologies, you should see 16 or 32 <u><span style="color: blue;">(Note: as kgulukota stated above, you should not see 32 alone in the SAM FLAG column)</span></u>, but not both. If you see both, or don't see either, your reads are paired strangely. &nbsp;(</span><a class="bigusername" href="http://seqanswers.com/forums/member.php?u=480" style="background-color: #e1e4f2; color: #ff4400; font-family: verdana, geneva, lucida, 'lucida grande', arial, helvetica, sans-serif; font-size: 13px; font-weight: bold;">swbarnes2</a>)<br /><br />Each reported read or pair alignment beyond the first has the SAM 'secondary' bit (which equals 256) set in its FLAG field. So, for multiple mapping reads, SAM alignments also contain their strand information. <br /><br />btw, For strand-specific RNAseq, the Tophat output SAM (converted from BAM) does not contain XS:A:+/- tag (which is required by cufflinks) for non-spliced reads. In order to get the proper strand info for the assembly, you need to manually add the tags. Here is an example code:<span style="background-color: white; color: #444444; font-family: Verdana, Arial, sans-serif; font-size: 13px; line-height: 20px;"><br /></span><br /><br /><span style="background-color: white; color: #444444; font-family: Verdana, Arial, sans-serif;"></span><br /><span style="background-color: white; color: #444444; font-family: 'Courier New', Courier, monospace;"><strike><span style="line-height: 20px;">samtools view -h accepted_hits.bam |&nbsp;</span><span style="background-color: white; line-height: 20px;">awk '{if($0 ~ /XS:A:/ || $1 ~ /^@/) print $0; else {if($2==16 || $2==272) print $0"\tXS:A:-"; if($2==0 || $2==256) print $0"\tXS:A:+";}}' accepted_hits.sam</span></strike></span><br /><br />FLAG=256 and 272 is the corresponding version of 0 and 16 for multiple mapped hits.<br /><br /><b>UPDATE: This only works for single-end lib. For pair-end lib, the FLAG should be odd number, but in any case, reads on minus strand always have 1 on the 5th bit of binary code (e.g. 0x10 =10000). Thanks to Wei's suggestion on this. Here is the updated code:</b><br /><br /><span style="color: #444444; font-family: 'Courier New', Courier, monospace; line-height: 20px;">samtools view -h accepted_hits.bam |&nbsp;</span><span style="background-color: white; color: #444444; font-family: 'Courier New', Courier, monospace; line-height: 20px;">awk '{if($0 ~ /XS:A:/ || $1 ~ /^@/) print $0; else {if(and($2,0x10)==16) print $0"\tXS:A:-"; else print $0"\tXS:A:+";}}' accepted_hits.sam</span><br /><span style="background-color: white; color: #444444; font-family: 'Courier New', Courier, monospace; line-height: 20px;"><br /></span><br /><b>UPDATE2: (from the&nbsp;</b><b>&nbsp;final version of&nbsp;</b><b>RNAseq lecture slides)</b><span style="font-family: Wingdings; text-align: left; text-indent: -0.31in;">Ø</span><span style="font-family: Arial; text-align: left; text-indent: -0.31in;">F</span><span style="font-family: Arial; text-align: left; text-indent: -0.31in;">or non-strand-specific lib, you’re actually sequencing </span><span style="font-family: Arial; text-align: left; text-indent: -0.31in;">cDNA</span><span style="font-family: Arial; text-align: left; text-indent: -0.31in;"> from both strands. So, the ‘strand’ info in the alignment is senseless (except for the spliced-reads).</span><br />      <!--StartFragment-->   <div style="direction: ltr; margin-bottom: 0pt; margin-left: .31in; margin-top: 0pt; text-align: left; text-indent: -.31in; unicode-bidi: embed; vertical-align: baseline;"><span style="font-family: Wingdings; mso-special-format: bullet;">Ø</span><span style="font-family: Arial;">For </span><span style="font-family: Arial;">strand</span><span style="font-family: Arial;">-specific lib, </span><span style="font-family: Arial;">if FLAG contains 0x10 (e.g. 0x53=0x40+</span><u><span style="font-family: Arial;">0x10</span></u><span style="font-family: Arial;">+0x2+0x1), </span><span style="font-family: Arial;">reads </span><span style="font-family: Arial;">map </span><span style="font-family: Arial;">to </span><span style="font-family: Arial;">‘minus</span><span style="font-family: Arial;">’ strand, </span><span style="font-family: Arial;">otherwise, ‘plus’ strand. </span></div><div style="direction: ltr; margin-bottom: 0pt; margin-left: .31in; margin-top: 0pt; text-align: left; text-indent: -.31in; unicode-bidi: embed; vertical-align: baseline;"><span style="font-family: Wingdings; mso-special-format: bullet;">Ø</span><span style="font-family: Arial;">Cufflinks requires XS:A</span><span style="font-family: Arial;">:+/- tag </span><span style="font-family: Arial;">(which </span><span style="font-family: Arial;">Tophat</span><span style="font-family: Arial;"></span><span style="font-family: Arial;">doesn</span><span style="font-family: Arial;">’</span><span style="font-family: Arial;">t have for some reads). </span><span style="font-family: Arial;">You need to manually add it </span><span style="font-family: Arial;">b</span><span style="font-family: Arial;">y command, e.g</span><span style="font-family: Arial;">. for </span><span style="font-family: Arial;"><a href="http://cufflinks.cbcb.umd.edu/manual.html">dUTP</a></span><span style="font-family: Arial;"> library (where /</span><span style="font-family: Arial;">2</span><span style="font-family: Arial;"> is from transcript strand, /1 from the opposite strand):</span></div><div style="direction: ltr; margin-bottom: 0pt; margin-left: .19in; margin-top: 0pt; text-align: left; unicode-bidi: embed; vertical-align: baseline;"><span style="font-family: 'Courier New'; font-size: 12pt;">samtools</span><span style="font-family: 'Courier New'; font-size: 12pt;"> view -h </span><span style="font-family: 'Courier New'; font-size: 12pt;">accepted_hits.bam</span><span style="font-family: 'Courier New'; font-size: 12pt;"> | </span><span style="font-family: 'Courier New'; font-size: 12pt;">awk</span><span style="font-family: 'Courier New'; font-size: 12pt;">'{if($0 ~ /XS:A:/ || $1 ~ /^@/) print $0; else {if(and($</span><span style="font-family: 'Courier New'; font-size: 12pt;">2,0x40) || and($2,0x90)) </span><span style="font-family: 'Courier New'; font-size: 12pt;">print $0"\</span><span style="font-family: 'Courier New'; font-size: 12pt;">tXS:A</span><span style="font-family: 'Courier New'; font-size: 12pt;">:-"; else print $0"\</span><span style="font-family: 'Courier New'; font-size: 12pt;">tXS:A</span><span style="font-family: 'Courier New'; font-size: 12pt;">:+";}}' </span><span style="font-family: 'Courier New'; font-size: 12pt;">accepted_hits.sam</span></div><!--EndFragment--><span style="background-color: white; color: #444444; font-family: 'Courier New', Courier, monospace; line-height: 20px;"><br /></span><br />&nbsp;See my post here:<br /><br /><a href="http://seqanswers.com/forums/showthread.php?p=69643#post69643">http://seqanswers.com/forums/showthread.php?p=69643#post69643</a><br /><br /><span style="background-color: #f5f5ff; font-family: verdana, geneva, lucida, 'lucida grande', arial, helvetica, sans-serif; font-size: 13px;">If you used a non-strand-specific protocol (which most people still do) you're not actually sequencing transcripts, you're sequencing cDNA with two strands. So the read could come from either of the two strands of a cDNA and you don't have any information which of the two strands corresponds to the original mRNA strand. This can be inferred when a read spans a splice junctions because splice site are highly conserved at the first 2 bases and last 2 bases of an intron. (</span><a class="bigusername" href="http://seqanswers.com/forums/member.php?u=2961" style="background-color: #e1e4f2; color: #ff4400; font-family: verdana, geneva, lucida, 'lucida grande', arial, helvetica, sans-serif; font-size: 13px; font-weight: bold;">Thomas Doktor</a><span style="background-color: #f5f5ff; font-family: verdana, geneva, lucida, 'lucida grande', arial, helvetica, sans-serif; font-size: 13px;">)</span><br /><span style="background-color: #f5f5ff; font-family: verdana, geneva, lucida, 'lucida grande', arial, helvetica, sans-serif; font-size: 13px;"><br /></span><br /># So, the strand info in the SAM alignment for non-strand-specific RNAseq lib cannot be used as evidence of transcript strand.