---
layout: post
title: How to tell which library type to use (fr-firststrand or fr-secondstrand)?
date: '2012-07-30T16:29:00.004-04:00'
author: Xianjun Dong
tags:
- next-generation sequencing
- UCSC
- Tophat
modified_time: '2012-07-31T09:56:11.105-04:00'
thumbnail: http://4.bp.blogspot.com/-Ocn91gy9E5M/UBc1emNFh9I/AAAAAAAAAVY/E6vmshsdeZ4/s72-c/Project+log+(BU_HD)+(1).png
blogger_id: tag:blogger.com,1999:blog-12445049.post-3881633812390575883
blogger_orig_url: http://onetipperday.blogspot.com/2012/07/how-to-tell-which-library-type-to-use.html
---

First of all, as a bioinformatian, you should ask the data producer (e.g. the one who prepared the RNAseq library) which&nbsp;protocol&nbsp;they used to generate the data.<br /><br />Tophat manual page has listed the general strand-specific protocol:<br /><br /><table cellspacing="15" style="background-color: white; color: #555555; font-family: Verdana, Arial, sans-serif; font-size: 13px; line-height: 20px; margin: 0px; padding: 0px;"><tbody style="margin: 0px; padding: 0px;"><tr style="margin: 0px; padding: 0px;"><td nowrap="nowrap" style="margin: 0px; padding: 0px;" valign="top"><strong style="margin: 0px; padding: 0px;">Library Type</strong></td><td nowrap="nowrap" style="margin: 0px; padding: 0px;" valign="top"><strong style="margin: 0px; padding: 0px;">Examples</strong></td><td style="margin: 0px; padding: 0px;" valign="top"><strong style="margin: 0px; padding: 0px;">Description</strong></td></tr><tr style="margin: 0px; padding: 0px;"><td nowrap="nowrap" style="margin: 0px; padding: 0px;" valign="top">fr-unstranded</td><td nowrap="nowrap" style="margin: 0px; padding: 0px;" valign="top"><tt style="font-family: 'Courier New', Courier, monospace; margin: 0px; padding: 0px;">Standard Illumina</tt></td><td style="margin: 0px; padding: 0px;" valign="top">Reads from the left-most end of the fragment (in transcript coordinates) map to the transcript strand, and the right-most end maps to the opposite strand.</td></tr><tr style="margin: 0px; padding: 0px;"><td nowrap="nowrap" style="margin: 0px; padding: 0px;" valign="top">fr-firststrand</td><td nowrap="nowrap" style="margin: 0px; padding: 0px;" valign="top"><tt style="font-family: 'Courier New', Courier, monospace; margin: 0px; padding: 0px;">dUTP, NSR, NNSR</tt></td><td style="margin: 0px; padding: 0px;" valign="top">Same as above except we enforce the rule that the right-most end of the fragment (in transcript coordinates) is the first sequenced (or only sequenced for single-end reads). Equivalently, it is assumed that only the strand generated during first strand synthesis is sequenced.</td></tr><tr style="margin: 0px; padding: 0px;"><td nowrap="nowrap" style="margin: 0px; padding: 0px;" valign="top">fr-secondstrand</td><td nowrap="nowrap" style="margin: 0px; padding: 0px;" valign="top"><tt style="font-family: 'Courier New', Courier, monospace; margin: 0px; padding: 0px;">Ligation, Standard SOLiD</tt></td><td style="margin: 0px; padding: 0px;" valign="top">Same as above except we enforce the rule that the left-most end of the fragment (in transcript coordinates) is the first sequenced (or only sequenced for single-end reads). Equivalently, it is assumed that only the strand generated during second strand synthesis is sequenced.</td></tr></tbody></table><br />In case you don't know the library-type, you can still figure it out by yourself. Tophat FAQ page provided a solution for that (http://tophat.cbcb.umd.edu/faq.html#library_type). But more simply (comparing to running 1M reads first), you can choose few reads and BLAT to genome and infer the library-type from the mapping result.<br /><br />Generally, reads from the left-most end of RNA fragment (always from 5´ to 3´) are always mapped to transcript-strand, and (for pair-end sequencing) reads from the right-most end are always mapped to the opposite strand. See the arrows direction in the below schema. This is because the sequencer always read from 5´ to 3´.<br /><div class="separator" style="clear: both; text-align: center;"></div><table align="center" cellpadding="0" cellspacing="0" class="tr-caption-container" style="margin-left: auto; margin-right: auto; text-align: center;"><tbody><tr><td style="text-align: center;"><a href="http://4.bp.blogspot.com/-Ocn91gy9E5M/UBc1emNFh9I/AAAAAAAAAVY/E6vmshsdeZ4/s1600/Project+log+(BU_HD)+(1).png" imageanchor="1" style="margin-left: auto; margin-right: auto;"><img border="0" src="http://4.bp.blogspot.com/-Ocn91gy9E5M/UBc1emNFh9I/AAAAAAAAAVY/E6vmshsdeZ4/s1600/Project+log+(BU_HD)+(1).png" /></a></td></tr><tr><td class="tr-caption" style="text-align: center;"><table align="center" cellpadding="0" cellspacing="0" class="tr-caption-container" style="margin-left: auto; margin-right: auto; text-align: center;"><tbody><tr><td class="tr-caption" style="font-size: 12.499999046325684px;">Summary of library type protocols (for Tophat/Bowtie)</td></tr></tbody></table></td></tr></tbody></table><br />But regarding to which strand the RNA fragment is synthesized from, this involves different strand-specific protocols. Thanks to the illustration figure (see below) from Zhao Zhang, we could see that for example dUTP method is to only sequence the strand from the first strand synthesis (the original RNA strand is &nbsp;degradated due to the dUTP incorporated), so the /2 read is from the&nbsp;original&nbsp;RNA strand.<br /><table align="center" cellpadding="0" cellspacing="0" class="tr-caption-container" style="margin-left: auto; margin-right: auto; text-align: center;"><tbody><tr><td style="text-align: center;"><a href="http://3.bp.blogspot.com/-BkupUsIrnXk/UBbmmmx6T8I/AAAAAAAAAUU/_rcrd_ahT48/s1600/strand.png" imageanchor="1" style="margin-left: auto; margin-right: auto; text-align: center;"><img border="0" height="445" src="http://3.bp.blogspot.com/-BkupUsIrnXk/UBbmmmx6T8I/AAAAAAAAAUU/_rcrd_ahT48/s640/strand.png" width="640" /></a></td></tr><tr><td class="tr-caption" style="text-align: center;">Strand-specific library protocols (Credit: Zhao Zhang)</td></tr></tbody></table>Taking a real example, first getting some reads (in fasta format)&nbsp;from the paired-end sequencing fastq file&nbsp;using command like:<br /><br />$ zcat ~/nearline/rnaseq/BU/Jul2012/Sample_3576_H_01.R1.fastq.gz | sed 's/@//g;s/ /_/g' | awk '{if(NR%4==1)print "&gt;"$0;if(NR%4==2) print $0;}' | head<br /><br />$ zcat ~/nearline/rnaseq/BU/Jul2012/Sample_3576_H_01.R2.fastq.gz | sed 's/@//g;s/ /_/g' | awk '{if(NR%4==1)print "&gt;"$0;if(NR%4==2) print $0;}' | head<br /><br />Blatting them in UCSC Genome Browser<br /><div class="separator" style="clear: both; text-align: center;"><a href="http://2.bp.blogspot.com/-CQPOg3cmirw/UBbuY8SFwQI/AAAAAAAAAU0/5Ao10X5ER18/s1600/Screen+shot+2012-07-30+at+4.27.10+PM.png" imageanchor="1" style="margin-left: 1em; margin-right: 1em;"><img border="0" height="299" src="http://2.bp.blogspot.com/-CQPOg3cmirw/UBbuY8SFwQI/AAAAAAAAAU0/5Ao10X5ER18/s640/Screen+shot+2012-07-30+at+4.27.10+PM.png" width="640" /></a></div><br />Below is screenshot for top hits of one pair of reads. They mapped to exons of OS9 genes (the left one is /1 and right one is /2, with opposite direction). We see that /1 mapped to transcript direction, /2 mapped to opposite direction, which means it can only be fr-secondstrand or fr-unstrand (cannot be fr-firststrand).<br /><div class="separator" style="clear: both; text-align: center;"></div><div class="separator" style="clear: both; text-align: center;"><a href="http://1.bp.blogspot.com/-4sjs7Cf-ox0/UBfjxdjVV1I/AAAAAAAAAVo/k4Ss08nAsgs/s1600/1.png" imageanchor="1" style="margin-left: 1em; margin-right: 1em;"><img border="0" src="http://1.bp.blogspot.com/-4sjs7Cf-ox0/UBfjxdjVV1I/AAAAAAAAAVo/k4Ss08nAsgs/s1600/1.png" /></a></div><div class="separator" style="clear: both; text-align: center;"><br /></div><div class="separator" style="clear: both; text-align: center;"></div><div class="separator" style="clear: both; text-align: center;"><br /></div>Continuing to look at other reads in the file, we can find examples like these:<br /><div class="separator" style="clear: both; text-align: center;"><a href="http://1.bp.blogspot.com/-BHQKBmDiv9M/UBfj2GRN1yI/AAAAAAAAAVw/JgPXiENaWmw/s1600/2.png" imageanchor="1" style="margin-left: 1em; margin-right: 1em;"><img border="0" src="http://1.bp.blogspot.com/-BHQKBmDiv9M/UBfj2GRN1yI/AAAAAAAAAVw/JgPXiENaWmw/s1600/2.png" /></a></div><div class="separator" style="clear: both; text-align: center;"><br /></div><div class="separator" style="clear: both; text-align: center;"><a href="http://4.bp.blogspot.com/-6G2io2-HduI/UBfj6jex_vI/AAAAAAAAAV4/EwrGQQydCyA/s1600/3.png" imageanchor="1" style="margin-left: 1em; margin-right: 1em;"><img border="0" src="http://4.bp.blogspot.com/-6G2io2-HduI/UBfj6jex_vI/AAAAAAAAAV4/EwrGQQydCyA/s1600/3.png" /></a></div>where /2 mapped to transcript strand and /1 mapped to the opposite strand. Combining with the observation from above, we can conclude that this is a fr-unstrand library.