---
layout: post
title: measuring expression of smallRNA genes
date: '2012-11-29T17:18:00.002-05:00'
author: Xianjun Dong
tags:
- cufflinks
- RPKM
- bedtools
- DEseq
- smallRNA
modified_time: '2012-11-30T15:16:09.277-05:00'
blogger_id: tag:blogger.com,1999:blog-12445049.post-7299095776150005927
blogger_orig_url: http://onetipperday.blogspot.com/2012/11/measuring-expression-of-smallrna-genes.html
---

As the <a href="http://onetipperday.blogspot.com/2012/11/dont-trust-cufflinks-fpkm-for-short.html" target="_blank">early note</a> mentioned, it's not a good idea to use bias-correction based method like cufflinks to calculate the RPKM/FPKM values for small RNA (simply because it's not designed for that purpose).<br /><br />Since the smallRNA is generally shorter, the usual RNA-seq protocol does not capture them properly (only the 'tail' of the distribution of fragment length). A good solution is to use smallRNA RNA-seq protocol. I meant using short reads length and without size selection. <br /><br />Based on that, we may be able to use tools like cufflinks/DESeq to calculate the RPKM/FPKM. But again, I am not sure cufflinks will still apply the bias-correction to the small RNA genes when the reads length is short enough. As this post discussed, "<a href="http://seqanswers.com/forums/showthread.php?t=6483">The Best way to normalize miRNA HTS data</a>", methods can be:<br /><br />- RPM ( Read Per Million ) <br />- RPKM (Reads per kilobase per million mapped) : Mortazi et al, Nat. Methods, 2008 (http://www.ncbi.nlm.nih.gov/pubmed/18516045)<br />- Trimmed mean of M-values : Robinson, Oshlack, Genome Biology 2010 (http://genomebiology.com/2010/11/3/R25)<br />- upper-quantile : http://www.ncbi.nlm.nih.gov/pubmed/20167110<br /><br />Simon (one of the authors of DESeq) suggested to use TMM or DESeq (which are reads count-based methods) for differential expression analysis, at least for small RNA genes.<br /><div><br /></div><div>As <a href="http://www.bioconductor.org/packages/devel/bioc/vignettes/DESeq/inst/doc/DESeq.pdf" target="_blank">the DEseq document</a>&nbsp;described, to get reads count you need alignment (SAM/BAM) and the annotation file as input for some tools, like</div><div><ul><li>the function <i>summarizeOverlaps</i> in the BioConductor <i>GenomicRanges</i> package,</li><li>the <i>geneCounts</i> function in the Bioconductor&nbsp;<i>easyRNASeq</i> package,</li><li>the <i>htseq-count</i> script distributed&nbsp;with the HTSeq Python package (Note:&nbsp;&nbsp;<i>htseq-count</i>&nbsp;produces one count file for each sample. DESeq offers the function <i>newCountDataSetFromHTSeqCount</i>&nbsp;to get an analysis started from these  files) &nbsp;</li><li>or the <i>bedtools multicov -count -bams sample1.bam&nbsp;<i>sample2</i>.bam&nbsp;<i>sample3</i>.bam -bed genes.bed</i></li></ul><div>Note the naming issue in Refseq or UCSC that same locus has been assigned with multiple gene names (e.g.&nbsp;<a href="http://genome.ucsc.edu/cgi-bin/hgTracks?hgHubConnect.destUrl=..%2Fcgi-bin%2FhgTracks&amp;clade=mammal&amp;org=Human&amp;db=hg19&amp;position=chr5%3A70220768-70248842&amp;hgt.positionInput=SMN2+%28Homo+sapiens+survival+of+motor+neuron+2%2C+centromeric+%28SMN2%29%2C+transcript+variant+d%2C+mRNA.%29&amp;hgt.suggestTrack=knownGene&amp;Submit=submit&amp;hgsid=311636999&amp;pix=1330&amp;hgFind.matches=uc003kak.3&amp;knownGene=pack" target="_blank">SMN1/2</a>), and htseq-count requires unique gene_id in the annotation. You might want to make sure of this before using htseq-count. One solution to this is to use Ensembl GTF.<br /><br />Also, be aware of the difference between bedtools multicov and htseq-count, esp. when you use multi-mappers.&nbsp;Read details here: <a href="http://seqanswers.com/forums/showthread.php?t=24903">http://seqanswers.com/forums/showthread.php?t=24903</a>.&nbsp;</div></div>