---
layout: post
title: Don't trust Cufflinks FPKM for short genes
date: '2012-11-29T13:08:00.003-05:00'
author: Xianjun Dong
tags:
- cufflinks
- RNAseq
modified_time: '2012-11-30T12:01:58.449-05:00'
blogger_id: tag:blogger.com,1999:blog-12445049.post-2216914344238584775
blogger_orig_url: http://onetipperday.blogspot.com/2012/11/dont-trust-cufflinks-fpkm-for-short.html
---

Here is what Cole (author of Cufflinks) commented to the&nbsp;observation of "<a href="http://seqanswers.com/forums/showthread.php?t=17404">very high RPKM values from Cufflink</a>":<br /><blockquote class="tr_bq">This issue has been discussed elsewhere on this board. As Nicholas points out, RNA-Seq really isn't reliable for very short transcripts. The reason is that all the fragments that map to these transcripts come from the "tail" of the distribution of library fragment lengths. That is, fragments that map to microRNAs are much, much shorter than most fragments in the library - by design in the RNA-Seq protocol, which size selects away very short inserts. Thus, <u>Cufflinks infers that even though relatively few fragments actually mapped to the microRNAs, there were probably TONS of individual microRNA molecules in the transcriptome before all of the various size selection parts of the protocol kicked in. Cufflinks accordingly increases the FPKM of these short transcripts to compensate for the bias against short fragments in the library.</u><br /><br />This compensation was designed to improve accuracy for transcripts that are in the 500bp-1kb range - for longer transcripts, the "edge effects" due to library fragment size aren't much of an issue. However, I wouldn't trust FPKM values for transcripts shorter than your average fragment length. There's really just not enough data in most standard RNA-Seq libraries to say much about small RNA abundance.<br />I should also point out that other methods use this same bias correction technique (RSEM for example). As far as I'm aware, the "count-based" methods don't, but that doesn't mean they shouldn't. <u>Most of those methods are strictly for differential analysis, where any edge effects are assumed to be affecting each condition the same way. </u>That may or may not be the case in your data.<br />In any case, the quick answer to this problem is to simply remove or ignore transcripts shorter than around 300bp from your GTF. In a future version, we will be flagging these transcripts as too short for reliable quantification where appropriate.</blockquote>Here is the full discussion: <a href="http://seqanswers.com/forums/archive/index.php/t-17404.html">http://seqanswers.com/forums/archive/index.php/t-17404.html</a>