---
layout: post
title: convert bed to gtf, gtf to bed
date: '2012-08-14T15:33:00.000-04:00'
author: Xianjun Dong
tags:
- GTF
- UCSC
- BED
- format converter
modified_time: '2014-02-10T10:09:02.310-05:00'
blogger_id: tag:blogger.com,1999:blog-12445049.post-9173920667605937815
blogger_orig_url: http://onetipperday.blogspot.com/2012/08/convert-bed-to-gtf.html
---

<h2>bed --&gt; gtf:</h2><div><span style="font-family: Courier New, Courier, monospace;"><b>bedToGenePred input.bed stdout | genePredToGtf file stdin output.gtf</b></span></div><div><br /></div>Here is an example from UCSC wiki (http://genomewiki.ucsc.edu/index.php/Genes_in_gtf_or_gff_format):<br /><br />Some gene tracks are in a bed format in the database, perhaps with extra columns past the standard bed format. In this case, extract the standard bed columns, convert it to a genePred and then to a gtf. For example <br /><br />mysql --user=genome --host=genome-mysql.cse.ucsc.edu -A -N -e "select chrom,chromStart,chromEnd,name,score,strand,thickStart,thickEnd from wgRna;" hg19 | bedToGenePred stdin stdout | genePredToGtf file stdin wgRna.gtf<br /><div style="margin-bottom: 0.5em; margin-top: 0.4em;"><div style="background-color: white;"><br /></div><h2 style="background-color: white;">gtf --&gt; bed:</h2><div style="background-color: white;">download gtf2bed.pl from Eric's site:</div><div style="background-color: white;">http://code.google.com/p/ea-utils/source/browse/trunk/clipper/gtf2bed</div><div style="background-color: white;"><span style="font-family: Courier New, Courier, monospace;"><b><br /></b></span></div><div style="background-color: white;"><span style="font-family: Courier New, Courier, monospace;"><b>perl&nbsp;</b></span><b style="font-family: 'Courier New', Courier, monospace;">gtf2bed.pl input.gtf &gt; out</b><b style="font-family: 'Courier New', Courier, monospace;">put.bed</b></div><div style="background-color: white;"><br /></div><div style="background-color: white;">If simply converting gtf to bed6 format (no exons/intron info), the following bash line can work:</div><div style="background-color: white;"><span style="font-family: Courier New, Courier, monospace;">fgrep -w transcript gencode.v17.annotation.gtf | sed 's/[";]//g;' | awk '{OFS="\t"; print $1, $4-4,$5,$12,0,$7}'</span></div>or a bed9 format with additional info on gene/type/name<br /><div style="background-color: white;"><span style="font-family: Courier New, Courier, monospace;">fgrep -w transcript gencode.v17.annotation.gtf | sed 's/[";]//g;' | awk '{OFS="\t"; print $1, $4-4,$5,$12,0,$7,$18,$14,$10}'&nbsp;</span></div><div style="background-color: white;"><br />Also, I edited Eric's code a bit to output a more meaningful gene ID, e,g.<br /><br /><span style="font-family: Courier New, Courier, monospace;">($transid) = $f[8]=~ /transcript_id "([^"]+)"/;</span><br /><span style="font-family: Courier New, Courier, monospace;">($geneid) = $f[8]=~ /gene_id "([^"]+)"/;</span><br /><span style="font-family: Courier New, Courier, monospace;">($gene_type) = $f[8]=~ /gene_type "([^"]+)"/;</span><br /><span style="font-family: Courier New, Courier, monospace;">($gene_name) = $f[8]=~ /gene_name "([^"]+)"/;</span><br /><span style="font-family: Courier New, Courier, monospace;">($trans_type) = $f[8]=~ /transcript_type "([^"]+)"/;</span><br /><span style="font-family: Courier New, Courier, monospace;">$id="${gene_name}__${geneid}__${transid}__${gene_type}.${trans_type}";</span></div></div>