---
layout: post
title: adaptor remover/clipper with mismatch allowed
date: '2012-03-14T16:55:00.000-04:00'
author: Xianjun Dong
tags:
- adaptor
- filter
- RNAseq
modified_time: '2012-03-15T14:52:41.023-04:00'
blogger_id: tag:blogger.com,1999:blog-12445049.post-4647203742853696372
blogger_orig_url: http://onetipperday.blogspot.com/2012/03/adaptor-removerclipper-with-mismatch.html
---

<br /><span style="font-family: 'Courier New', Courier, monospace;">$ head -n12 RNAseq/mouse_adult_wt_smallRNAseq_76nt_strand_ox.fastq&nbsp;</span><br /><span style="font-family: 'Courier New', Courier, monospace;">@HWUSI-EAS1533_0026_FC:6:1:1204:14021#0/1</span><br /><span style="font-family: 'Courier New', Courier, monospace;">TTCCTATCAATAAATCTCATGTGANNGCAGC<span style="background-color: yellow;">TCGTANNCCNTCTNCNNNT</span><span style="background-color: yellow;">NN</span>NNANNNNANANCNNNNNANNNNNN</span><br /><span style="font-family: 'Courier New', Courier, monospace;">+HWUSI-EAS1533_0026_FC:6:1:1204:14021#0/1</span><br /><span style="font-family: 'Courier New', Courier, monospace;">gggaegggggfdggfcaedbBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBB</span><br /><span style="font-family: 'Courier New', Courier, monospace;">@HWUSI-EAS1533_0026_FC:6:1:1204:19652#0/1</span><br /><span style="font-family: 'Courier New', Courier, monospace;">TGGTATAAACTCTTCTAAAAGACTNNAGC<span style="background-color: yellow;">TCGTATGNNG</span><b style="background-color: red;">C</b><span style="background-color: yellow;">NTTCNGNNNG</span>NNNNANNNNANANANNNNNANNNNNN</span><br /><span style="font-family: 'Courier New', Courier, monospace;">+HWUSI-EAS1533_0026_FC:6:1:1204:19652#0/1</span><br /><span style="font-family: 'Courier New', Courier, monospace;">c^^[^a`ddYaddadf_YafSOTZDDYZK\M`W_`BBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBB</span><br /><span style="font-family: 'Courier New', Courier, monospace;">@HWUSI-EAS1533_0026_FC:6:1:1205:19274#0/1</span><br /><span style="font-family: 'Courier New', Courier, monospace;">TCAGAAATGTCCTGTGAAACTCGCNNAATT<span style="background-color: yellow;">TCGTATGCCGCCTTCTGNNTG</span>NAAANNNAAAACAANNNNANNNNNN</span><br /><span style="font-family: 'Courier New', Courier, monospace;">+HWUSI-EAS1533_0026_FC:6:1:1205:19274#0/1</span><br /><span style="font-family: 'Courier New', Courier, monospace;">fV^V]b]`ZSTXdRbc]dac`BBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBB</span><br /><br /><br /><div>The basic question is to trim/clip the part after the adaptor sequence (e.g. TCGTATGCCGTCTTCTGCTTG in above example) in the FASTQ reads. It's easy if there is no mismatch. For case with mismatch, here is few solutions I got from google:</div><div><ol><li><b>FAR</b> (http://sourceforge.net/apps/mediawiki/theflexibleadap/index.php?title=Main_Page), which uses Needleman-Wunsch alignment (global alignment) to align the adaptor sequence(s) to the read sequences and find the first best alignment.<br /><br /><span style="font-family: 'Courier New', Courier, monospace;">far -s stdin.fastq -t stdout.fq -f&nbsp;<span style="background-color: white; line-height: 19px;">fastq-sanger</span>&nbsp;-as TCGTATGCCGTCTTCTGCTTG --cut-off 5 --min-overlap 15 --phred-pre-trim 20 --min-readlength 20 --trim-end right --adaptive-overlap yes --max-uncalled 30&nbsp;--nr-threads 8&nbsp;--log-level ALL</span></li><li>&nbsp;<b>Scythe</b> only checks for 3'-end contaminants, up to the adapter's length into the 3'-end. For reads with contamination in any position, the program <b>TagDust</b> (<a href="http://genome.gsc.riken.jp/osc/english/dataresource/">http://genome.gsc.riken.jp/osc/english/dataresource/</a>) is recommended. Scythe has the advantages of allowing fuzzier matching and being base quality-aware, while TagDust has the advantages of very fast matching (but allowing few mismatches, and not considering quality) and FDR. TagDust also removes contaminated reads entirely, while Scythe trims off contaminants.</li><br />A possible pipeline would run FASTQ reads through Scythe, then TagDust, then a quality-based trimmer, and finally through a read quality statistics program such as qrqc (<a href="http://bioconductor.org/packages/devel/bioc/html/qrqc.html">http://bioconductor.org/packages/devel/bioc/html/qrqc.html</a>) or FASTqc (<a href="http://www.bioinformatics.bbsrc.ac.uk/projects/fastqc/">http://www.bioinformatics.bbsrc.ac.uk/projects/fastqc/</a>). <li><b>trimLRPatterns</b>() function in BioConductor. Here is an&nbsp;<a href="http://manuals.bioinformatics.ucr.edu/home/ht-seq#TOC-Trimming-of-Flanking-Sequences-e.g.">example</a>.</li><li>Another very useful toolkit is fastx_toolkit, which includes a set of tools for handling fasta and fastq format.&nbsp;<b><a href="http://hannonlab.cshl.edu/fastx_toolkit/commandline.html#fastx_clipper_usage">fastx_clipper</a> &nbsp;</b>is for clipping adaptor, <strike>but from what I checked it does not allow mismatch (e.g. N is counted as 'match' in fastx_clipper, but in FAR it's mismatch)</strike>. &nbsp;Running the following code on above reads only clips the last read. The first one should also be clipped. Don't know why. (Just get reply from Gordon that fastx_clipper does allow mismatch and also count 'N' as mismatch. Just the first two sequences have high error rate (=number_of_mismatch/adaptor_alignment_length, e.g. 33% in this case), so it's not counted as an adaptor.&nbsp;</li></ol><div><div><span style="font-family: 'Courier New', Courier, monospace;">$ head -n12 RNAseq/mouse_adult_wt_smallRNAseq_76nt_strand_ox.fastq | fastx_clipper -a TCGTATGCCGTCTTCTGCTTG -c -n -i -&nbsp;</span></div><div><span style="font-family: 'Courier New', Courier, monospace;">@HWUSI-EAS1533_0026_FC:6:1:1205:19274#0/1</span></div><div><span style="font-family: 'Courier New', Courier, monospace;">TCAGAAATGTCCTGTGAAACTCGCNNAATT</span></div><div><span style="font-family: 'Courier New', Courier, monospace;">+HWUSI-EAS1533_0026_FC:6:1:1205:19274#0/1</span></div><div><span style="font-family: 'Courier New', Courier, monospace;">fV^V]b]`ZSTXdRbc]dac`BBBBBBBBB</span></div></div></div>