---
layout: post
title: get intron, UTR, CDS from bed12 format
date: '2012-11-19T12:22:00.000-05:00'
author: Xianjun Dong
tags:
- BED12
- format converter
modified_time: '2015-09-29T11:18:06.260-04:00'
blogger_id: tag:blogger.com,1999:blog-12445049.post-1699977216330915005
blogger_orig_url: http://onetipperday.blogspot.com/2012/11/get-intron-utr-cds-from-bed12-format.html
---

Input: gene annotation in&nbsp;BED12 format (If your input is GTF/GFF, you want to convert into bed12 ahead, <a href="http://onetipperday.blogspot.com/2012/08/convert-bed-to-gtf.html">see my another post for howto</a>)<br />Output: intron/UTR/CDS in bed12 format (which can be further converted into bed6 via "<span style="font-family: Courier New, Courier, monospace;">bedtools bed12tobed6</span>" command)<br /><br /><b># Introns (if any) of gene annotation in BED12 format</b><br />cat annotation.bed | awk '{OFS="\t";split($11,a,","); split($12,b,","); A=""; B=""; for(i=1;i&lt;length(a)-1;i++) {A=A""(b[i+1]-b[i]-a[i])",";B=B""(b[i]+a[i]-(b[1]+a[1]))",";} if($10&gt;1) print $1,$2+a[1], $3-a[length(a)-1], $4,$5,$6,$2+a[1], $3-a[length(a)-1],$9,$10-1,A,B;}'<br /><br /><b># 5´ UTR (if any) of gene annotation in BED12 format</b><br />cat annotation.bed |&nbsp;awk '{OFS="\t";split($11,a,","); split($12,b,","); A=""; B=""; &nbsp;if($7==$8) next; if($6=="+" &amp;&amp; $2&lt;$7) {for(i=1;i&lt;length(a);i++) if(($2+b[i]+a[i])&lt;=$7) {A=A""a[i]",";B=B""b[i]",";} else {A=A""($7-$2-b[i])",";B=B""b[i]","; break; } print $1,$2,$7,$4,$5,$6,$2,$7,$9,i,A,B;} if($6=="-" &amp;&amp; $8&lt;$3) {for(i=length(a)-1;i&gt;0;i--) if(($2+b[i])&gt;=$8) {A=a[i]","A;B=($2+b[i]-$8)","B;} else {A=($2+b[i]+a[i]-$8)","A;B=0","B; break; } print $1,$8,$3,$4,$5,$6,$8,$3,$9,length(a)-i,A,B;}}'<br /><br /><b># CDS (if any) of gene annotation in BED12 format</b><br /><span style="color: red;">grep "protein_coding\.protein_coding"</span>&nbsp;annotation.bed |&nbsp;awk '{OFS="\t";split($11,a,","); split($12,b,","); A=""; B=""; if($7==$8) next; j=0; for(i=1;i&lt;length(a);i++) if(($2+b[i]+a[i])&gt;$7 &amp;&amp; ($2+b[i])&lt;$8) {j++; start=$2+b[i]-$7; size=a[i]; if(($2+b[i])&lt;=$7) {start=0;size=size-($7-($2+b[i]));} if(($2+a[i]+b[i])&gt;=$8) {size=size-($2+a[i]+b[i]-$8);} A=A""size",";B=B""start",";} print $1,$7,$8,$4,$5,$6,$7,$8,$9,j,A,B;}'<br /><br />### Note1: The thickStart and thickEnd in the bed12 format don't always indicate CDS. "<span style="background-color: white;">When there is no thick part, thickStart and thickEnd are usually set to the chromStart position</span>." <span style="background-color: yellow;">So, we cannot use bed12 to infer CDS (or coding exons), esp. for lincRNA</span>. The correct way is to grep all CDS lines from the GTF file, or directly using UCSC Table Browser to download coding exons. <br /><br />### Note2: The "<span style="background-color: yellow;">Polymorphic pseudogene</span>" in GENCODE can also have a protein-coding transcript, even though the gene itself is classified as a pseudogene (YES, polymorphic pseudogenes are also pseudogenes, they "are coding gene that are pseudogenic due to the presence of a polymorphic premature stop codon in the reference genome" (<a href="http://www.genomebiology.com/2012/13/9/R51">http://www.genomebiology.com/2012/13/9/R51</a>). For more details, see <a href="https://gencodegenes.wordpress.com/toolbox">https://gencodegenes.wordpress.com/toolbox</a>/.         <br /><br /><b># 3´ UTR (if any) of gene annotation in BED12 format</b><br />cat annotation.bed |&nbsp;awk '{OFS="\t";split($11,a,","); split($12,b,","); A=""; B=""; if($7==$8) next; if($6=="-" &amp;&amp; <span style="color: red;">$2&lt;$7</span>) {for(i=1;i&lt;length(a);i++) if(($2+b[i]+a[i])&lt;=$7) {A=A""a[i]",";B=B""b[i]",";} else {A=A""($7-$2-b[i])",";B=B""b[i]","; break; } print $1,$2,$7,$4,$5,$6,$2,$7,$9,i,A,B;} if($6=="+" &amp;&amp; <span style="color: red;">$8&lt;$3</span>) {for(i=length(a)-1;i&gt;0;i--) if(($2+b[i])&gt;=$8) {A=a[i]","A;B=($2+b[i]-$8)","B;} else {A=($2+b[i]+a[i]-$8)","A;B=0","B; break; } print $1,$8,$3,$4,$5,$6,$8,$3,$9,length(a)-i,A,B;}}'<br /><br />## NOTE: The red part was switched for 3'UTR extraction. It's now fixed.