---
layout: post
title: Three ways to trim adaptor/primer sequences for paired-end reads
date: '2012-08-01T13:41:00.001-04:00'
author: Xianjun Dong
tags:
- next-generation sequencing
modified_time: '2012-08-09T21:51:04.110-04:00'
thumbnail: http://1.bp.blogspot.com/-eDCkz2A8OGQ/UBiQqnsRx3I/AAAAAAAAAWI/jhwIDlK7eQ0/s72-c/Screen+shot+2012-07-31+at+10.12.29+PM.png
blogger_id: tag:blogger.com,1999:blog-12445049.post-2911646923493619874
blogger_orig_url: http://onetipperday.blogspot.com/2012/08/three-ways-to-trim-adaptorprimer.html
---

<h3><b>1. Understanding the adaptors</b> <span style="font-weight: normal;"><span style="font-size: small;">(skip this part if you're familiar with the Illumina adaptor)</span></span></h3>Before trimming anything from the reads, let's get clear what the reads content is.<br /><br />Taking Trufseq reads (from Illumina HiSeq 2000) as example, here is the read file (fastq) looks like:<br /><br /><span style="font-family: 'Courier New', Courier, monospace; font-size: x-small;"><b>$ cat r1.fq&nbsp;</b></span><br /><span style="font-family: 'Courier New', Courier, monospace; font-size: x-small;">@3VFXHS1:278:D13Y4ACXX:1:1101:1472:2209 1:N:0:<b>CGATGT</b></span><br /><span style="font-family: 'Courier New', Courier, monospace; font-size: x-small;">CTGGTATTGTCTCTTCCCACACTGAACTCTGGGGAATT<b>CGATGT</b>GTGGCACAGCCCGGCTCAGCCTGCCCGCTGGTGGGAGCCCCTGGGAAGCTGCGGCGC</span><br /><span style="font-family: 'Courier New', Courier, monospace; font-size: x-small;">+</span><br /><span style="font-family: 'Courier New', Courier, monospace; font-size: x-small;">@@CFDDFFGH&gt;CAEH:CGHIJJJJEIHJJHIJJJ?DHIDIJHGEGHJG;FHC9@B(5@6A=EH:B@B@2=&gt;&gt;B?BDCBD&lt;B52&lt;&lt;ABD?&lt;?B1@A9&gt;B###</span><br /><span style="font-family: 'Courier New', Courier, monospace; font-size: x-small;">@3VFXHS1:278:D13Y4ACXX:1:1101:1434:2224 1:N:0:CGATGT</span><br /><span style="font-family: 'Courier New', Courier, monospace; font-size: x-small;">GGCAGAGCCAATCTTCGGACGTGGTGATTGTCTCCTCTAAGTACAAACAGCGCTATGAGTGTCGCCTGCCAGCTGGAGCTATTCACTTCCAGCGTGAAAGG</span><br /><span style="font-family: 'Courier New', Courier, monospace; font-size: x-small;">+</span><br /><span style="font-family: 'Courier New', Courier, monospace; font-size: x-small;">BC@FFFFFHHHGFHIIJJIJGFHICFCGIHGFHFGGCHD@F?B?BGGHJJIG6D@EHEHHEHCD259?AACD@AC59?,(5&gt;A,;&gt;:@C(::(029?8&gt;@A</span><br /><span style="font-family: 'Courier New', Courier, monospace; font-size: x-small;">@3VFXHS1:278:D13Y4ACXX:1:1101:1712:2247 1:N:0:CGATGT</span><br /><span style="font-family: 'Courier New', Courier, monospace; font-size: x-small;">GTACACTTGAACACATTTTTCTAACCTTAGAAAATACCTACAAGGCCTGTTGTCTTGACCCATTACTCAATTGTCCCTGGCATATTATCTGATCTTCACGT</span><br /><span style="font-family: 'Courier New', Courier, monospace; font-size: x-small;">+</span><br /><span style="font-family: 'Courier New', Courier, monospace; font-size: x-small;">CCCFFFFFHHHHGJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJIJGIIJGGGDHHHIHIHHEIJIJJIIGGIIIIFEHHHHFEFFFDEEEDCEEEDEDC?A</span><br /><span style="font-family: 'Courier New', Courier, monospace; font-size: x-small;">@3VFXHS1:278:D13Y4ACXX:1:1101:3318:2215 1:N:0:<b>CGATGT</b></span><br /><span style="font-family: 'Courier New', Courier, monospace; font-size: x-small;">A<span style="color: red;">GATCGGAAGAGCACACGTCTGAACTCCAGTCAC<b>CGATGT</b>ATCTCGTATGCCGTCTTCTGCTTG</span>AAAAAAAAAAGAAAAAAAAACAAGCGACAAGGACAGA</span><br /><span style="font-family: 'Courier New', Courier, monospace; font-size: x-small;">+</span><br /><span style="font-family: 'Courier New', Courier, monospace; font-size: x-small;">CCCFFFFFHHHHHJJJJJIJJJJJJJJJIJGHFFHIAHIFGGIIJJIJJFIJIHJIHHHGEGFE&gt;CFFEB###############################</span><br /><span style="font-family: 'Courier New', Courier, monospace; font-size: x-small;">@3VFXHS1:278:D13Y4ACXX:2:1101:5344:2243 1:N:0:<b>ACAGTG</b></span><br /><span style="font-size: x-small;"><span style="color: red; font-family: 'Courier New', Courier, monospace;">GATCGGAAGAGCACACGTCTGAACTCCAGTCAC<b>ACAGTG</b>ATCTCGTATGCCGTCTTCTGCTTG</span><span style="font-family: 'Courier New', Courier, monospace;">AAAAAAAAACAACAGAAAAAACAAAGCGCGAACAGTGC</span></span><br /><span style="font-family: 'Courier New', Courier, monospace; font-size: x-small;">+</span><br /><span style="font-family: 'Courier New', Courier, monospace; font-size: x-small;">CCCFFFFFHHHHHJIJJIJIJHJJJJJJJHIGIGIHJFHFFDIIJGIIIJJJHHFIJJJJJJHCEHHFD################################</span><br /><span style="font-family: 'Courier New', Courier, monospace;"><br /></span><span style="font-family: 'Courier New', Courier, monospace; font-size: x-small;"><b>$ cat r2.fq&nbsp;</b></span><br /><span style="font-family: 'Courier New', Courier, monospace; font-size: x-small;">@3VFXHS1:278:D13Y4ACXX:1:1101:1472:2209 2:N:0:CGATGT</span><br /><span style="font-family: 'Courier New', Courier, monospace; font-size: x-small;">CTGGATTTGAAATCTTTAGCGGAGCGGGAACGCCGGCGCGGAAGGGTCTCTACACAGGGCCCGGTCCGCCCTTGCGCTCTCCTTAATGNNNNNNNNNNCGC</span><br /><span style="font-family: 'Courier New', Courier, monospace; font-size: x-small;">+</span><br /><span style="font-family: 'Courier New', Courier, monospace; font-size: x-small;">@CCF?EFFHHHGHHHIFGI@HGGIEHIGIJGI6@@E&gt;B8&gt;??:DBD++399&gt;ACCDDDD@DBDD58@BBDDDDD@@&lt;@BDDDDC&gt;CDC#############</span><br /><span style="font-family: 'Courier New', Courier, monospace; font-size: x-small;">@3VFXHS1:278:D13Y4ACXX:1:1101:1434:2224 2:N:0:CGATGT</span><br /><span style="font-family: 'Courier New', Courier, monospace; font-size: x-small;">CCCGGGGCCTCCCATTAAGGTCGCACTTGGACCCATTGCCATAGGTCTGGCTGTGGTAGCGTTTAAGACGATGCTGCTTGGAGGCCTTGGCTGTTTCATCA</span><br /><span style="font-family: 'Courier New', Courier, monospace; font-size: x-small;">+</span><br /><span style="font-family: 'Courier New', Courier, monospace; font-size: x-small;">BCCFFFFDHFFHGIHIHGJJIJJGGIIJJJIDHGJIJEIIIIIJJIIJIJHHHEEF;&gt;DDA&gt;BBB@CAABBBDDCDDDDDDAD@@?CDDDCCB?ACCDDC#</span><br /><span style="font-family: 'Courier New', Courier, monospace; font-size: x-small;">@3VFXHS1:278:D13Y4ACXX:1:1101:1712:2247 2:N:0:CGATGT</span><br /><span style="font-family: 'Courier New', Courier, monospace; font-size: x-small;">GTTACTCAGCATTTATTCATGCCTGCTGTGTACGGAAAGGGCAGTTACAAAGGAAAGCCTTGATGATTCTGCTTCCAAGAAACGTGAAGATCAGATAATAT</span><br /><span style="font-family: 'Courier New', Courier, monospace; font-size: x-small;">+</span><br /><span style="font-family: 'Courier New', Courier, monospace; font-size: x-small;">CBCFFFFFHHHHHJIJJIIJJJJJIIJIHIHIJJJIJIJIJJIIIIIIJJGGIIJJJIJJJJHIJJJJHIJHGHHHHFFFFFEDECBDDDDDCCDDDCDEE</span><br /><span style="font-family: 'Courier New', Courier, monospace; font-size: x-small;">@3VFXHS1:278:D13Y4ACXX:1:1101:3318:2215 2:N:0:CGATGT</span><br /><span style="font-family: 'Courier New', Courier, monospace; font-size: x-small;"><span style="color: red;">AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT</span>AAAAAAAAAAAAAAAACAGAAAAAGAAAGAAGGAAAGGNGAGT</span><br /><span style="font-family: 'Courier New', Courier, monospace; font-size: x-small;">+</span><br /><span style="font-family: 'Courier New', Courier, monospace; font-size: x-small;">CCCFFFFFHHHHHJJJJFHHHJJJIJJGGIDGHIEIJJJIIJGIJ5@FHIB?DFFEEEDEEEDD?####################################</span><br /><span style="font-family: 'Courier New', Courier, monospace; font-size: x-small;">@3VFXHS1:278:D13Y4ACXX:2:1101:5344:2243 2:N:0:ACAGTG</span><br /><span style="font-family: 'Courier New', Courier, monospace; font-size: x-small;">GGATCGGGAAAGGGGGGGGGGGGGGAAAAGGGGGGATTTCCGGGGGGGCCGGTTCTTTTAAAAAAAAAAAAAAAGAAAACAGAAACAGAAGATGGACAACA</span><br /><span style="font-family: 'Courier New', Courier, monospace; font-size: x-small;">+</span><br /><span style="font-family: 'Courier New', Courier, monospace; font-size: x-small;">CCCFFFFFHHHHHJJJJFHHHJJJIJJGGIDGHIEIJJJIIJGIJ5@FHIB?DFFEEEDEEEDD?####################################</span><br /><br /><div>First of first, it's critical to understand what your reads file contain; do they contain adaptor sequences? do they contain primer sequence? I strongly recommend to read the description file here:&nbsp;http://genomics.med.tufts.edu/documents/protocols/TUCF_Understanding_Illumina_TruSeq_Adapters.pdf, from which we could know that the constructed dsDNA (before binding to the flow cell for sequencing) looks like:&nbsp;</div><div class="separator" style="clear: both; text-align: center;"><a href="http://1.bp.blogspot.com/-eDCkz2A8OGQ/UBiQqnsRx3I/AAAAAAAAAWI/jhwIDlK7eQ0/s1600/Screen+shot+2012-07-31+at+10.12.29+PM.png" imageanchor="1" style="margin-left: 1em; margin-right: 1em;"><img border="0" height="156" src="http://1.bp.blogspot.com/-eDCkz2A8OGQ/UBiQqnsRx3I/AAAAAAAAAWI/jhwIDlK7eQ0/s640/Screen+shot+2012-07-31+at+10.12.29+PM.png" width="640" /></a></div><div>Where&nbsp;</div><div><br /></div><div><b id="internal-source-marker_0.7076293954160064"><span style="background-color: #eeeeee; font-family: 'Courier New'; font-size: 15px; font-weight: normal; vertical-align: baseline; white-space: pre-wrap;">Trufseq Universal Adaptor:</span><br /><span style="background-color: #eeeeee; font-family: 'Courier New'; font-size: 15px; font-weight: normal; vertical-align: baseline; white-space: pre-wrap;">5´AATGATACGGCGACCACCGAGATCTACACTCTTTCCCTACACGAC</span><span style="background-color: red; font-family: 'Courier New'; font-size: 15px; font-weight: bold; vertical-align: baseline; white-space: pre-wrap;">GCTCTTCCGATC</span><span style="background-color: #eeeeee; font-family: 'Courier New'; font-size: 15px; font-weight: normal; vertical-align: baseline; white-space: pre-wrap;">T3´</span></b></div><div><span style="font-family: 'Courier New';"><span style="font-size: 14.999999046325684px; white-space: pre-wrap;">--&gt; reverse complementary</span></span></div><div><b id="internal-source-marker_0.7076293954160064" style="font-weight: normal;"><span style="background-color: #eeeeee; font-family: 'Courier New'; font-size: 15px; vertical-align: baseline; white-space: pre-wrap;">5´A</span><span style="background-color: red; font-family: 'Courier New'; font-size: 15px; font-weight: bold; vertical-align: baseline; white-space: pre-wrap;">GATCGGAAGAGC</span><span style="background-color: #eeeeee; font-family: 'Courier New'; font-size: 15px; vertical-align: baseline; white-space: pre-wrap;">GTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT3´</span></b><span style="font-family: 'Courier New';"><span style="font-size: 14.999999046325684px; white-space: pre-wrap;"><br /></span></span><br /><span style="background-color: #eeeeee; font-family: 'Courier New'; font-size: 15px; font-weight: normal; vertical-align: baseline; white-space: pre-wrap;">TruSeq Indexed Adapter:</span><br /><span style="background-color: #eeeeee; font-family: 'Courier New'; font-size: 15px; font-weight: normal; vertical-align: baseline; white-space: pre-wrap;">5´</span><span style="background-color: red; font-family: 'Courier New'; font-size: 15px; font-weight: bold; vertical-align: baseline; white-space: pre-wrap;">GATCGGAAGAGC</span><span style="background-color: #eeeeee; font-family: 'Courier New'; font-size: 15px; font-weight: normal; vertical-align: baseline; white-space: pre-wrap;">ACACGTCTGAACTCCAGTCAC‐NNNNNN‐ATCTCGTATGCCGTCTTCTGCTTG3´</span></div><div><br /></div><div>The 6-nt "<b id="internal-source-marker_0.7076293954160064"><span style="background-color: #eeeeee; font-family: 'Courier New'; font-size: 15px; font-weight: normal; vertical-align: baseline; white-space: pre-wrap;">NNNNNN</span></b>" is barcode for multiplexing. We noticed that the 3´ of Universal adaptor is reverse-complementary to the 5´ of Indexed adaptor (Why? This is to form the Y-shape adaptor. See ZZ's dUTP figure in another post).&nbsp;</div><div><br /></div>Combining together the "Overrepresented sequences" of FASTQC output file (<a href="http://zlab.umassmed.edu/~dongx/tracks/RNAseq/BU/Sample_3576_H_01.R1_fastqc/fastqc_report.html#M9">for example</a>), we could infer that the reads are contaminated by adaptor/primers (for example the red-marked part in the fastq sequences).<br /><div><br /></div><div>So, next step is how to remove the contamination.&nbsp;</div><div><h3><b>2. Collect adaptor sequences</b></h3></div><div>To remove contamination, we first should collect all possible "contamination" sources. For our case, we collect all used barcode sequences and generated the adaptor file:<br /><div><br /></div><div><div>http://zlab.umassmed.edu/~dongx/tracks/RNAseq/BU/Demultiplex_Stats.htm</div><div>copy content and save to txt file&nbsp;Demultiplex_Stats.htm</div><div>grep -v Undetermined barcode_stat.html | awk '{print "&gt;"$2; print "GATCGGAAGAGCACACGTCTGAACTCCAGTCAC"$4"ATCTCGTATGCCGTCTTCTGCTTG";}' &gt; adaptor.fa</div><div><br /></div><div>The&nbsp;Demultiplex_Stats.htm file, which contains barcode information of each sample, is usually included in the output folder of sequencing. Otherwise, you can consult from the data producer.&nbsp;</div><div><br /></div><div>You may also want to append the universal adaptor:</div><div>$cat &gt;&gt; adaptor.fa</div><div>&gt;Trufseq_Universal_Adaptor</div>AATGATACGGCGACCACCGAGATCTACACTCTTTCCCTACACGACGCTCTTCCGATCT</div><div>(Ctrl+D)</div><div><br /><div><div>A full list of commonly-used adaptors can be retrieved from the following URL (returned by Google search) by command:</div><div><br /></div><div>curl -s http://www.omicsoft.com/downloads/ngs/contamination_list/v1.txt 2&gt;&amp;1 | sed "/^\s\+$/d;s/\t\+/\t/g;s/ /_/g;s/,//g;s/'//g" | awk '{print "&gt;"$1; print $2;}' &gt; adaptors_list.fa&nbsp;</div></div><div><br /></div><div>This does not include Trufseq barcodes used for smallRNA (see below), which you have to include yourself:</div><div>http://epigenome.usc.edu/docs/resources/core_protocols/TruSeq%20index%20sequences-3.pdf</div><div><h3><b>3. Remove adaptor&nbsp;</b></h3></div><div>There might be many ways to remove adaptors, but specifically for PE reads (e.g. both reads are removed if one of the pairs is disqualified), I'd like to introduce three ways to do this.</div><div><br /></div><div><b>a. FastqMcf (<a href="http://code.google.com/p/ea-utils/wiki/FastqMcf" style="font-family: inherit;">http://code.google.com/p/ea-utils/wiki/FastqMcf</a>)</b></div><div><div><span class="pln" style="font-family: Monaco, 'DejaVu Sans Mono', 'Bitstream Vera Sans Mono', 'Lucida Console', monospace; font-size: 12px;"><br /></span></div><span style="font-family: 'Courier New', Courier, monospace;">usage: fastq-mcf [options] &lt;adapters.fa&gt; &lt;reads.fq&gt; [mates1.fq ...]</span><br /><div><span style="font-family: 'Courier New', Courier, monospace;"><br /></span></div><div><span style="font-family: 'Courier New', Courier, monospace;">Options:</span></div><div><span style="font-family: 'Courier New', Courier, monospace;">&nbsp; &nbsp; -h &nbsp; &nbsp; &nbsp;This help</span></div><div><span style="font-family: 'Courier New', Courier, monospace;">&nbsp; &nbsp; -o FIL &nbsp;Output file (stats to stdout)</span></div><div><span style="font-family: 'Courier New', Courier, monospace;">&nbsp; &nbsp; -s N.N &nbsp;Log scale for clip pct to threshold (2.2)</span></div><div><span style="font-family: 'Courier New', Courier, monospace;">&nbsp; &nbsp; -t N &nbsp; &nbsp;% occurance threshold before clipping (0.25)</span></div><div><span style="font-family: 'Courier New', Courier, monospace;">&nbsp; &nbsp; -m N &nbsp; &nbsp;Minimum clip length, overrides scaled auto (1)</span></div><div><span style="font-family: 'Courier New', Courier, monospace;">&nbsp; &nbsp; -p N &nbsp; &nbsp;Maximum adapter difference percentage (10)</span></div><div><span style="font-family: 'Courier New', Courier, monospace;">&nbsp; &nbsp; -l N &nbsp; &nbsp;Minimum remaining sequence length (19)</span></div><div><span style="font-family: 'Courier New', Courier, monospace;">&nbsp; &nbsp; -L N &nbsp; &nbsp;Maximum sequence length (none)</span></div><div><span style="font-family: 'Courier New', Courier, monospace;">&nbsp; &nbsp; -k N &nbsp; &nbsp;sKew percentage-less-than causing trim (2)</span></div><div><span style="font-family: 'Courier New', Courier, monospace;">&nbsp; &nbsp; -q N &nbsp; &nbsp;quality threshold causing trimming (10)</span></div><div><span style="font-family: 'Courier New', Courier, monospace;">&nbsp; &nbsp; -w N &nbsp; &nbsp;window-size for quality trimming (1)</span></div><div><span style="font-family: 'Courier New', Courier, monospace;">&nbsp; &nbsp; -f &nbsp; &nbsp; &nbsp;force output, even if not much will be done</span></div><div><span style="font-family: 'Courier New', Courier, monospace;">&nbsp; &nbsp; -F FIL &nbsp;remove sequences that align to FIL</span></div><div><span style="font-family: 'Courier New', Courier, monospace;">&nbsp; &nbsp; -0 &nbsp; &nbsp; &nbsp;Set all trimming parameters to zero</span></div><div><span style="font-family: 'Courier New', Courier, monospace;">&nbsp; &nbsp; -U|u &nbsp; &nbsp;Force disable/enable illumina PF filtering</span></div><div><span style="font-family: 'Courier New', Courier, monospace;">&nbsp; &nbsp; -P N &nbsp; &nbsp;phred-scale (auto)</span></div><div><span style="font-family: 'Courier New', Courier, monospace;">&nbsp; &nbsp; -x N &nbsp; &nbsp;'N' (Bad read) percentage causing trimming (20)</span></div><div><span style="font-family: 'Courier New', Courier, monospace;">&nbsp; &nbsp; -R &nbsp; &nbsp; &nbsp;Don't remove N's from the fronts/ends of reads</span></div><div><span style="font-family: 'Courier New', Courier, monospace;">&nbsp; &nbsp; -n &nbsp; &nbsp; &nbsp;Don't clip, just output what would be done</span></div><div><span style="font-family: 'Courier New', Courier, monospace;">&nbsp; &nbsp; -C N &nbsp; &nbsp;Number of reads to use for subsampling (200k)</span></div><div><span style="font-family: 'Courier New', Courier, monospace;">&nbsp; &nbsp; -S FIL &nbsp;Save clipped reads to file</span></div><div><span style="font-family: 'Courier New', Courier, monospace;">&nbsp; &nbsp; -d &nbsp; &nbsp; &nbsp;Output lots of random debugging stuff</span></div></div><div><br /></div><div>For example,&nbsp;</div><div>fastq-mcf -o c1.fq -o c2.fq&nbsp;-l 16 -q 15 -w 4 -x 10 -u -P 33 adaptor.fa r1.fq r2.fq &amp;&gt;r.log</div><div><br /></div><div><b>b. Trimmomatic&nbsp;(<a href="http://www.usadellab.org/cms/index.php?page=trimmomatic">http://www.usadellab.org/cms/index.php?page=trimmomatic</a>)</b></div><br />Paired End Mode:<br />java -classpath &lt;path to trimmomatic jar&gt; org.usadellab.trimmomatic.TrimmomaticPE [-threads &lt;threads&gt;] [-phred33 | -phred64] [-trimlog &lt;logFile&gt;] &lt;input 1&gt; &lt;input 2&gt; &lt;paired output 1&gt; &lt;unpaired output 1&gt; &lt;paired output 2&gt; &lt;unpaired output 2&gt; &lt;step 1&gt; ...<br /><div><br />Step options:<br /><ul><li>ILLUMINACLIP:&lt;fastaWithAdaptersEtc&gt;:&lt;seed mismatches&gt;:&lt;palindrome clip threshold&gt;:&lt;simple clip threshold&gt;</li><ul><li>fastaWithAdaptersEtc: specifies the path to a fasta file containing all the adapters, PCR sequences etc. The naming of the various sequences within this file determines how they are used. See below.</li><li>seedMismatches: specifies the maximum mismatch count which will still allow a full match to be performed</li><li>palindromeClipThreshold: specifies how accurate the match between the two 'adapter ligated' reads must be for PE palindrome read alignment.</li><li>simpleClipThreshold: specifies how accurate the match between any adapter etc. sequence must be against a read.</li></ul><li>SLIDINGWINDOW:&lt;windowSize&gt;:&lt;requiredQuality&gt;</li><ul><li>windowSize: specifies the number of bases to average across</li><li>requiredQuality: specifies the average quality required.</li></ul><li>LEADING:&lt;quality&gt;</li><ul><li>quality: Specifies the minimum quality required to keep a base.</li></ul><li>TRAILING:&lt;quality&gt;</li><ul><li>quality: Specifies the minimum quality required to keep a base.</li></ul><li>CROP:&lt;length&gt;</li><ul><li>length: The number of bases to keep, from the start of the read.</li></ul><li>HEADCROP:&lt;length&gt;</li><ul><li>length: The number of bases to remove from the start of the read.</li></ul><li>MINLENGTH:&lt;length&gt;</li><ul><li>length: Specifies the minimum length of reads to be kept.</li></ul></ul>Following the above example:</div><div><br /></div><div>java -classpath $CLASSPATH/trimmomatic-0.22.jar org.usadellab.trimmomatic.TrimmomaticPE -phred33&nbsp;-trimlog r.log&nbsp;r1.fq r2.fq t1.fq t1.unpaired.fq t2.fq t2.unpaired.fq LEADING:3 TRAILING:3 ILLUMINACLIP:adaptor.fa:2:40:15 SLIDINGWINDOW:4:15 MINLEN:16&nbsp;</div><div><div><br />These two programs give same results:</div><div><br /></div><div><div><span style="font-family: 'Courier New', Courier, monospace; font-size: x-small;">@3VFXHS1:278:D13Y4ACXX:1:1101:1472:2209 1:N:0:CGATGT</span></div><div><span style="font-family: 'Courier New', Courier, monospace; font-size: x-small;">CTGGTATTGTCTCTTCCCACACTGAACTCTGGGGAATTCGATGTGTGGCACAGCCCGGCTCAGCCTGCCCGCTGGTGGGAGCCCCTGGGAAGCTGCGG</span></div><div><span style="font-family: 'Courier New', Courier, monospace; font-size: x-small;">+</span></div><div><span style="font-family: 'Courier New', Courier, monospace; font-size: x-small;">@@CFDDFFGH&gt;CAEH:CGHIJJJJEIHJJHIJJJ?DHIDIJHGEGHJG;FHC9@B(5@6A=EH:B@B@2=&gt;&gt;B?BDCBD&lt;B52&lt;&lt;ABD?&lt;?B1@A9&gt;B</span></div><div><span style="font-family: 'Courier New', Courier, monospace; font-size: x-small;">@3VFXHS1:278:D13Y4ACXX:1:1101:1434:2224 1:N:0:CGATGT</span></div><div><span style="font-family: 'Courier New', Courier, monospace; font-size: x-small;">GGCAGAGCCAATCTTCGGACGTGGTGATTGTCTCCTCTAAGTACAAACAGCGCTATGAGTGTCGCCTGCCAGCTGGAGCTATTCACTTCCAGCGTGAAAGG</span></div><div><span style="font-family: 'Courier New', Courier, monospace; font-size: x-small;">+</span></div><div><span style="font-family: 'Courier New', Courier, monospace; font-size: x-small;">BC@FFFFFHHHGFHIIJJIJGFHICFCGIHGFHFGGCHD@F?B?BGGHJJIG6D@EHEHHEHCD259?AACD@AC59?,(5&gt;A,;&gt;:@C(::(029?8&gt;@A</span></div><div><span style="font-family: 'Courier New', Courier, monospace; font-size: x-small;">@3VFXHS1:278:D13Y4ACXX:1:1101:1712:2247 1:N:0:CGATGT</span></div><div><span style="font-family: 'Courier New', Courier, monospace; font-size: x-small;">GTACACTTGAACACATTTTTCTAACCTTAGAAAATACCTACAAGGCCTGTTGTCTTGACCCATTACTCAATTGTCCCTGGCATATTATCTGATCTTCACGT</span></div><div><span style="font-family: 'Courier New', Courier, monospace; font-size: x-small;">+</span></div><div><span style="font-family: 'Courier New', Courier, monospace; font-size: x-small;">CCCFFFFFHHHHGJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJIJGIIJGGGDHHHIHIHHEIJIJJIIGGIIIIFEHHHHFEFFFDEEEDCEEEDEDC?A</span></div></div></div><div><span style="font-family: 'Courier New', Courier, monospace;"><br /></span></div><div><span style="font-weight: bold;">c.&nbsp;</span><b>HTSeq</b><span style="font-weight: bold;">&nbsp;</span><br /><span style="font-weight: bold;">(</span><a href="http://www-huber.embl.de/users/anders/HTSeq/doc/sequences.html#HTSeq.SequenceWithQualities.trim_left_end_with_quals">http://www-huber.embl.de/users/anders/HTSeq/doc/sequences.html#HTSeq.SequenceWithQualities.trim_left_end_with_quals</a><span style="font-weight: bold;">)</span></div><div><span style="font-weight: bold;"><br /></span></div><div>Cite what Simon posted <a href="http://seqanswers.com/forums/showthread.php?t=12171" target="_blank">here</a>, HTSeq has facilities useful for this.<br /><br />You can write a little Python script like this:<br /><blockquote class="tr_bq"><span style="font-family: Courier New, Courier, monospace;">import itertools</span><br /><span style="font-family: Courier New, Courier, monospace;">import HTSeq</span><br /><span style="font-family: Courier New, Courier, monospace;">in1 = iter( HTSeq.FastqReader( "mydata_1.fastq" ) )</span><br /><span style="font-family: Courier New, Courier, monospace;">in2 = iter( HTSeq.FastqReader( "mydata_2.fastq" ) )</span><br /><span style="font-family: Courier New, Courier, monospace;">out1 = open( "trimmed_1.fastq", "w" )</span><br /><span style="font-family: Courier New, Courier, monospace;">out2 = open( "trimmed_2.fastq", "w" )</span><br /><span style="font-family: Courier New, Courier, monospace;">for read1, read2 in itertools.izip( in1, in2 ):</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; &nbsp;read1.trim_right_end( "ACGGTC" )</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; &nbsp;read2.trim_left_end( "TTCGAC" )</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; &nbsp;read1.write_to_fastq_file( out1 )</span><br /><span style="font-family: Courier New, Courier, monospace;">&nbsp; &nbsp;read2.write_to_fastq_file( out2 )</span><br /><span style="font-family: Courier New, Courier, monospace;">out1.close()</span><br /><span style="font-family: Courier New, Courier, monospace;">out2.close()</span></blockquote></div><div>I've not figured out how to apply a fasta file of adaptor.&nbsp;</div></div></div>