---
layout: post
title: quick ways to extract mRNA/transcript sequence for a gene
date: '2013-01-25T10:08:00.001-05:00'
author: Xianjun Dong
tags:
- GTF
- mRNA
- BED
- data manipulation
modified_time: '2013-03-11T00:32:43.769-04:00'
blogger_id: tag:blogger.com,1999:blog-12445049.post-647268062721021196
blogger_orig_url: http://onetipperday.blogspot.com/2013/01/quick-ways-to-extract-mrnatranscript.html
---

<b><span style="color: red;">NOTE: please be aware that&nbsp;gtf_to_fasta tool in Tophat does NOT consider the strand info in input gff.</span>&nbsp;</b><br /><b><br /></b><b>1. Use "twoBitToFa -bed"</b><br />Download the <u>latest</u> twoBitToFa from Jim Kent's source, e.g. http://hgdownload.cse.ucsc.edu/admin/exe/linux.x86_64/twoBitToFa (link for linux version)<br /><br />Example:<br />twoBitToFa&nbsp;-bed=input.bed&nbsp;/path/input.2bit&nbsp;inputbed.cDNA.fa<br /><br />Here is the full usage:<br /><br />twoBitToFa - Convert all or part of .2bit file to fasta<br />usage:<br />&nbsp; &nbsp;twoBitToFa input.2bit output.fa<br />options:<br />&nbsp; &nbsp;-seq=name - restrict this to just one sequence<br />&nbsp; &nbsp;-start=X &nbsp;- start at given position in sequence (zero-based)<br />&nbsp; &nbsp;-end=X - end at given position in sequence (non-inclusive)<br />&nbsp; &nbsp;-seqList=file - file containing list of the desired sequence names<br />&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; in the format seqSpec[:start-end], e.g. chr1 or chr1:0-189<br />&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; where coordinates are half-open zero-based, i.e. [start,end)<br />&nbsp; &nbsp;-noMask - convert sequence to all upper case<br />&nbsp; &nbsp;-bpt=index.bpt - use bpt index instead of built in one<br />&nbsp; &nbsp;-bed=input.bed - grab sequences specified by input.bed. <u>Will exclude introns</u><br /><br />Sequence and range may also be specified as part of the input<br />file name using the syntax:<br />&nbsp; &nbsp; &nbsp; /path/input.2bit:name<br />&nbsp; &nbsp;or<br />&nbsp; &nbsp; &nbsp; /path/input.2bit:name<br />&nbsp; &nbsp;or<br />&nbsp; &nbsp; &nbsp; /path/input.2bit:name:start-end<br /><div><br /></div><br />To use the program, you need 2bit file for your genome. To convert fasta file to 2bit, use&nbsp;faToTwoBit (http://hgdownload.cse.ucsc.edu/admin/exe/linux.x86_64/faToTwoBit). It also requires your gene annotation in bed12 format.<br /><br /><b>2. Use "gffread -w" in Cufflinks utility (</b>http://cufflinks.cbcb.umd.edu/gff.html)<br /><br />Example:<br />gffread -w transcripts.fa -g /path/to/genome.fa transcripts.gtf<br /><br />where the input annotation can be gtf or gff2/3.<br /><div><br /></div><div>Note that this tool is powerful in only extracting the CDS/protein etc. And to get fast, it's better to have genome.fa.fai index file in the same path.<br /><div><br />Here is full usage:<br /><br />Usage:<br />gffread &lt;input_gff&gt; [-g &lt;genomic_seqs_fasta&gt; | &lt;dir&gt;][-s &lt;seq_info.fsize&gt;]<br />&nbsp;[-o &lt;outfile.gff&gt;] [-t &lt;tname&gt;] [-r [[&lt;strand&gt;]&lt;chr&gt;:]&lt;start&gt;..&lt;end&gt; [-R]]<br />&nbsp;[-CTVNJMKQAFGUBHZWTOLE] [-w &lt;exons.fa&gt;] [-x &lt;cds.fa&gt;] [-y &lt;tr_cds.fa&gt;]<br />&nbsp;[-i &lt;maxintron&gt;]<br />&nbsp;Filters and/or converts GFF3/GTF2 records.<br />&nbsp;&lt;input_gff&gt; is a GFF file, use '-' if the GFF records will be given at stdin<br /><br />&nbsp;Options:<br />&nbsp; -g &nbsp;full path to a multi-fasta file with the genomic sequences<br />&nbsp; &nbsp; &nbsp; for all input mappings, OR a directory with single-fasta files<br />&nbsp; &nbsp; &nbsp; (one per genomic sequence, with file names matching sequence names)<br />&nbsp; -s &nbsp;&lt;seq_info.fsize&gt; is a tab-delimited file providing this info<br />&nbsp; &nbsp; &nbsp; for each of the mapped sequences:<br />&nbsp; &nbsp; &nbsp; &lt;seq-name&gt; &lt;seq-length&gt; &lt;seq-description&gt;<br />&nbsp; &nbsp; &nbsp; (useful for -A option with mRNA/EST/protein mappings)<br />&nbsp; -i &nbsp;discard transcripts having an intron larger than &lt;maxintron&gt;<br />&nbsp; -r &nbsp;only show transcripts overlapping coordinate range &lt;start&gt;..&lt;end&gt;<br />&nbsp; &nbsp; &nbsp; (on chromosome/contig &lt;chr&gt;, strand &lt;strand&gt; if provided)<br />&nbsp; -R &nbsp;for -r option, discard all transcripts that are not fully<br />&nbsp; &nbsp; &nbsp; contained within the given range<br />&nbsp; -U &nbsp;discard single-exon transcripts<br />&nbsp; -C &nbsp;coding only: discard mRNAs that have no CDS feature<br />&nbsp; -F &nbsp;full GFF attribute preservation (all attributes are shown)<br />&nbsp; -G &nbsp;only parse additional exon attributes from the first exon<br />&nbsp; &nbsp; &nbsp; and move them to the mRNA level (useful for GTF input)<br />&nbsp; -A &nbsp;use the description field from &lt;seq_info.fsize&gt; and add it<br />&nbsp; &nbsp; &nbsp; as the value for a 'descr' attribute to the GFF record<br /><br />&nbsp; -O &nbsp;process also non-transcript GFF records (by default non-transcript<br />&nbsp; &nbsp; &nbsp; records are ignored)<br />&nbsp; -V &nbsp;discard any mRNAs with CDS having in-frame stop codons<br />&nbsp; -H &nbsp;for -V option, check and adjust the starting CDS phase<br />&nbsp; &nbsp; &nbsp; if the original phase leads to a translation with an<br />&nbsp; &nbsp; &nbsp; in-frame stop codon<br />&nbsp; -B &nbsp;for -V option, single-exon transcripts are also checked on the<br />&nbsp; &nbsp; &nbsp; opposite strand<br />&nbsp; -N &nbsp;discard multi-exon mRNAs that have any intron with a non-canonical<br />&nbsp; &nbsp; &nbsp; splice site consensus (i.e. not GT-AG, GC-AG or AT-AC)<br />&nbsp; -J &nbsp;discard any mRNAs that either lack initial START codon<br />&nbsp; &nbsp; &nbsp; or the terminal STOP codon, or have an in-frame stop codon<br />&nbsp; &nbsp; &nbsp; (only print mRNAs with a fulll, valid CDS)<br /><br />&nbsp; -M/--merge : cluster the input transcripts into loci, collapsing matching<br />&nbsp; &nbsp; &nbsp; &nbsp;transcripts (those with the same exact introns and fully contained)<br />&nbsp; -d &lt;dupinfo&gt; : for -M option, write collapsing info to file &lt;dupinfo&gt;<br />&nbsp; --cluster-only: same as --merge but without collapsing matching transcripts<br />&nbsp; -K &nbsp;for -M option: also collapse shorter, fully contained transcripts<br />&nbsp; &nbsp; &nbsp; with fewer introns than the container<br />&nbsp; -Q &nbsp;for -M option, remove the containment restriction:<br />&nbsp; &nbsp; &nbsp; (multi-exon transcripts will be collapsed if just their introns match,<br />&nbsp; &nbsp; &nbsp; while single-exon transcripts can partially overlap (80%))<br /><br />&nbsp; -E &nbsp;expose (warn about) duplicate transcript IDs and other potential<br />&nbsp; &nbsp; &nbsp; problems with the given GFF/GTF records<br />&nbsp; -Z &nbsp;merge close exons into a single exon (for intron size&lt;4)<br />&nbsp; -w &nbsp;write a fasta file with spliced exons for each GFF transcript<br />&nbsp; -x &nbsp;write a fasta file with spliced CDS for each GFF transcript<br />&nbsp; -W &nbsp;for -w and -x options, also write for each fasta record the exon<br />&nbsp; &nbsp; &nbsp; coordinates projected onto the spliced sequence<br />&nbsp; -y &nbsp;write a protein fasta file with the translation of CDS for each record<br />&nbsp; -L &nbsp;Ensembl GTF to GFF3 conversion (implies -F; should be used with -m)<br />&nbsp; -m &nbsp;&lt;chr_replace&gt; is a reference (genomic) sequence replacement table with<br />&nbsp; &nbsp; &nbsp; this format:<br />&nbsp; &nbsp; &nbsp; &lt;original_ref_ID&gt; &lt;new_ref_ID&gt;<br />&nbsp; &nbsp; &nbsp; GFF records on reference sequences that are not found among the<br />&nbsp; &nbsp; &nbsp; &lt;original_ref_ID&gt; entries in this file will be filtered out<br />&nbsp; -o &nbsp;the "filtered" GFF records will be written to &lt;outfile.gff&gt;<br />&nbsp; &nbsp; &nbsp; (use -o- for printing to stdout)<br />&nbsp; -t &nbsp;use &lt;trackname&gt; in the second column of each GFF output line<br />&nbsp; -T &nbsp;-o option will output GTF format instead of GFF3</div></div>