---
layout: post
title: How to select MACS peaks based on p-value, fold_enrichment and FDR?
date: '2013-08-08T16:16:00.001-04:00'
author: Xianjun Dong
tags:
- Poisson distribution
- statistics
- next-generation sequencing
- ChIPseq
- MACS
modified_time: '2013-08-13T10:49:10.403-04:00'
thumbnail: http://2.bp.blogspot.com/-nwXPkrhwuOo/UgPOEcGm7tI/AAAAAAAAAis/ITce1fiX3e0/s72-c/MACS.png
blogger_id: tag:blogger.com,1999:blog-12445049.post-7533042498777413849
blogger_orig_url: http://onetipperday.blogspot.com/2013/08/how-to-select-macs-peaks-based-on-p.html
---

<div class="separator" style="clear: both; text-align: left;">In the output peaks.xls file from MACS(v1.4), there are 3 columns: -10log10(pvalue), fold_enrichment, and FDR(%). We can use these three columns to sort/filter peaks. Here are what they means:</div><blockquote class="tr_bq">The '−10*log10(Pvalue)' column lists the transformed P value of each peak, which makes peak sorting easier. For example, a P value of 1e−5 would be transformed to 50. The 'fold_enrichment' column shows the ratio of <u>the ChIP-seq read count</u> to <u>the local value of lambda</u> within each peak. The 'FDR(%)' column contains the empirical FDR percentage for each peak. For example, the fourth peak in the list has an 'FDR(%)' value of '6.45' and '−10*log10(Pvalue)' value of '83.67'; using the same P value cutoff of 4.4e−09 = 1083.67/−10, the ratio of the number of peaks identified by MACS after and before exchanging control and ChIP-seq samples is 6.45:100. The FDR column is only available when the control sample is available.</blockquote><div class="separator" style="clear: both; text-align: left;">As the MACS paper (<i>Zhang et al., Genome Biology, 2008</i>) clearly stated, MACS uses Poisson distribution to measure the distribution of ChIP-seq tag. <a href="http://en.wikipedia.org/wiki/Poisson_distribution">Poisson distribution</a>&nbsp;is characterized as having only one parameter, λ, for both mean and variance. The origin of using Poisson distribution to measure ChIPseq tag can be referred to a <i>Nature</i> paper (see its <a href="http://www.nature.com/nature/journal/v448/n7153/extref/nature06008-s1.pdf">supplementary note</a>):</div><div><blockquote class="tr_bq">The number of sequence reads required to map a chromatin feature can be estimated from a&nbsp;simple model.<br />Suppose that the genome is divided into <i>N</i> non-overlapping bins of fixed size, that a fraction <i>f&nbsp;</i>of these bins contain a particular chromatin feature and that one performs ChIP-Seq with an&nbsp;antibody that enriches the sequence in these bins by a factor of <i>e</i>. If one collects a total of <i>R</i>&nbsp;sequence reads, the number of reads in a bin should approximately follow a Poisson&nbsp;distribution with mean <i>eM</i> for bins containing the feature and <i>M</i> for the other bins, where <i>M =&nbsp;</i><i>R/N(ef+(1-f))</i>.&nbsp;</blockquote><span style="background-color: #fff2cc;"><b>Xianjun</b>: For those who have difficulty to understand the formula, I would suggest to just understand it like this: if there is a single Poisson distribution, the λ should be R/N; when there are two parts/distribution, we just divide N into Nf and N(1-f) and give different weight to them. For the unenriched part, you want to weight the enriched bins with weight e (because it's more enriched). That is R/(Nfe+N(1-f))=M; For the enriched part, you want to weight unenriched bins with 1/e, then the λ is R/(Nf+N(1-f)/e)=eM. (Thanks to Shikui and Sowmya for helping understanding Poisson distribution and the formula!)</span><br /><br /><div>MACS uses a dynamic λ to compensate the local fluctuations and biases even in control sample:&nbsp;</div><div><span style="font-family: inherit;"><br /></span><br /><div style="background-color: white; border: 0px; display: table; line-height: 11.11111068725586px; margin: 0px; outline: 0px; padding: 0px; vertical-align: baseline; width: 567.361083984375px;"><span style="font-family: inherit;">λ<sub style="margin: 0px; padding: 0px;">local&nbsp;</sub>= max(λ<sub style="margin: 0px; padding: 0px;">BG</sub>, [λ<sub style="margin: 0px; padding: 0px;">1k</sub>,] λ<sub style="margin: 0px; padding: 0px;">5k</sub>, λ<sub style="margin: 0px; padding: 0px;">10k</sub>)</span></div><div style="background-color: white; border: 0px; clear: both; line-height: 1.6em; margin-bottom: 1em; outline: 0px; overflow: visible; padding: 0px; vertical-align: baseline;"></div><div style="border: 0px; clear: both; margin-bottom: 1em; outline: 0px; overflow: visible; padding: 0px; vertical-align: baseline;"><span style="font-family: inherit; line-height: 17.36111068725586px;"><span style="background-color: white;">where</span><span style="background-color: white; line-height: 17.36111068725586px;">&nbsp;λ</span><sub style="background-color: white; line-height: 17.36111068725586px; margin: 0px; padding: 0px;">BG </sub><span style="background-color: white;">is a uniform&nbsp;</span><span style="background-color: white; line-height: 17.36111068725586px;">estimation for the whole genome,&nbsp;</span><span style="background-color: white; line-height: 17.36111068725586px;">λ</span><sub style="background-color: white; line-height: 17.36111068725586px; margin: 0px; padding: 0px;">1k</sub><span style="background-color: white; line-height: 17.36111068725586px;">, λ</span><sub style="background-color: white; line-height: 17.36111068725586px; margin: 0px; padding: 0px;">5k&nbsp;</sub><span style="background-color: white; line-height: 17.36111068725586px;">and λ</span><sub style="background-color: white; line-height: 17.36111068725586px; margin: 0px; padding: 0px;">10k&nbsp;</sub><span style="background-color: white; line-height: 17.36111068725586px;">are λ estimated from the 1 kb, 5 kb or 10 kb window centered at the peak location in the control sample, or the ChIP-Seq sample when a control sample is not available (in which case λ</span><sub style="background-color: white; line-height: 17.36111068725586px; margin: 0px; padding: 0px;">1k&nbsp;</sub><span style="background-color: white; line-height: 17.36111068725586px;">is not used). λ</span><sub style="background-color: white; line-height: 17.36111068725586px; margin: 0px; padding: 0px;">local&nbsp;</sub><span style="line-height: 17.36111068725586px;"><span style="background-color: white;">captures the influence of local biases, and is robust against occasional low tag counts at small local regions.</span><span style="background-color: #fff2cc;"> <b>Xianjun</b>: That's why you will get different p-value and fold_enrichment for the same peak when using different control samples. </span><span style="background-color: white;">&nbsp;</span><span style="background-color: white;">MACS uses λ</span></span><span style="background-color: white;"><sub style="line-height: 17.36111068725586px; margin: 0px; padding: 0px;">local&nbsp;</sub><span style="line-height: 17.36111068725586px;">to calculate the&nbsp;</span><em style="border: 0px; line-height: 17.36111068725586px; margin: 0px; outline: 0px; padding: 0px; vertical-align: baseline;">p</em></span><span style="line-height: 17.36111068725586px;"><span style="background-color: white;">-value of each candidate peak (</span><span style="background-color: #fff2cc;"><b>Xianjun</b>: Once you have&nbsp;</span></span></span><span style="background-color: #fff2cc;"><span style="line-height: 17.36111068725586px;">λ, you have the Poisson distribution; once you have&nbsp;</span><span style="font-family: inherit;"><span style="line-height: 17.36111068725586px;">a distribution and the observed value, which is the observed reads count in the peak region for this case, then you can get p-value based on the distribution. See wikipedia for </span></span><a href="http://en.wikipedia.org/wiki/File:P-value_Graph.png" style="font-family: inherit; line-height: 17.36111068725586px;">p-value</a><span style="font-family: inherit;"><span style="line-height: 17.36111068725586px;">&nbsp;</span></span><span style="line-height: 17.359375px;">computation</span></span><span style="background-color: white; font-family: inherit;"><span style="line-height: 17.36111068725586px;">) and removes potential false positives due to local biases (that is, peaks significantly under λ</span></span><sub style="background-color: white; font-family: inherit; line-height: 17.36111068725586px; margin: 0px; padding: 0px;">BG</sub><span style="background-color: white; font-family: inherit; line-height: 17.36111068725586px;">, but not under λ</span><sub style="background-color: white; font-family: inherit; line-height: 17.36111068725586px; margin: 0px; padding: 0px;">local</sub><span style="background-color: white; font-family: inherit; line-height: 17.36111068725586px;">). Candidate peaks with&nbsp;</span><em style="background-color: white; border: 0px; font-family: inherit; line-height: 17.36111068725586px; margin: 0px; outline: 0px; padding: 0px; vertical-align: baseline;">p</em><span style="background-color: white; font-family: inherit; line-height: 17.36111068725586px;">-values below a user-defined threshold&nbsp;</span><em style="background-color: white; border: 0px; font-family: inherit; line-height: 17.36111068725586px; margin: 0px; outline: 0px; padding: 0px; vertical-align: baseline;">p</em><span style="background-color: white; font-family: inherit; line-height: 17.36111068725586px;">-value (default 10</span><sup style="background-color: white; font-family: inherit; line-height: 17.36111068725586px; margin: 0px; padding: 0px;">-5</sup><span style="background-color: white; font-family: inherit; line-height: 17.36111068725586px;">) are called, and the ratio between the ChIP-Seq tag count and λ</span><sub style="background-color: white; font-family: inherit; line-height: 17.36111068725586px; margin: 0px; padding: 0px;">local&nbsp;</sub><span style="background-color: white; font-family: inherit; line-height: 17.36111068725586px;">is reported as the&nbsp;</span><em style="background-color: white; border: 0px; font-family: inherit; line-height: 17.36111068725586px; margin: 0px; outline: 0px; padding: 0px; vertical-align: baseline;">fold_enrichment</em><span style="background-color: white; font-family: inherit; line-height: 17.36111068725586px;">.&nbsp;</span></div><div class="separator" style="clear: both; text-align: left;"><br /></div><div class="separator" style="clear: both; text-align: left;"><span style="background-color: yellow;">The question is: how to select MACS peaks based on the measurements?</span></div><div class="separator" style="clear: both; text-align: left;"><br /></div><div class="separator" style="clear: both; text-align: left;">To illustrate the relationship of the three measurements, I plot them as below. Note: the figures below are generated by Excel. Looking not so nice, but only for illustration purpose.&nbsp;</div><div class="separator" style="clear: both; text-align: left;"><br /></div><div class="separator" style="clear: both; text-align: center;"><a href="http://2.bp.blogspot.com/-nwXPkrhwuOo/UgPOEcGm7tI/AAAAAAAAAis/ITce1fiX3e0/s1600/MACS.png" imageanchor="1" style="margin-left: 1em; margin-right: 1em;"><img border="0" height="402" src="http://2.bp.blogspot.com/-nwXPkrhwuOo/UgPOEcGm7tI/AAAAAAAAAis/ITce1fiX3e0/s640/MACS.png" width="640" /></a></div><br />X-axis: -10log10(p-value); Y-axis: FDR (%)<br /><div class="separator" style="clear: both; text-align: center;"><br /></div><div class="separator" style="clear: both; text-align: center;"><a href="http://1.bp.blogspot.com/-IPL8QgYYAao/UgPQJ52of3I/AAAAAAAAAjA/g4MGIcEr3S0/s1600/MACS3.png" imageanchor="1" style="margin-left: 1em; margin-right: 1em;"><img border="0" height="392" src="http://1.bp.blogspot.com/-IPL8QgYYAao/UgPQJ52of3I/AAAAAAAAAjA/g4MGIcEr3S0/s640/MACS3.png" width="640" /></a></div>X-axis: fold_enrichment; &nbsp;Y-axis: FDR (%)<br /><div><br /></div>I don't know why FDR has a functional relationship with the p-value (<a href="https://groups.google.com/forum/#!topic/macs-announcement/MWy-9fXx8pQ/discussion">Neither Tao</a>).<br /><br />But we can choose a smaller FDR cutoff (e.g. 5%), since <a href="http://onetipperday.blogspot.com/2013/08/methods-of-calling-differential-region.html">it's underestimated due to a small control&nbsp;size in our case</a>. That would correspond to a p-value of 10^-50, significant enough.<br /><br />Here is the fold_enrichment histogram if we choose FDR cutoff as 10%.<br /><div class="separator" style="clear: both; text-align: center;"><a href="http://4.bp.blogspot.com/-TRxFC2D7QQ0/UgP9QwtAF2I/AAAAAAAAAjw/wtP2MI8Rfos/s1600/Screen+shot+2013-08-08+at+4.18.34+PM.png" imageanchor="1" style="margin-left: 1em; margin-right: 1em;"><img border="0" src="http://4.bp.blogspot.com/-TRxFC2D7QQ0/UgP9QwtAF2I/AAAAAAAAAjw/wtP2MI8Rfos/s1600/Screen+shot+2013-08-08+at+4.18.34+PM.png" /></a></div><div class="separator" style="clear: both; text-align: center;"><br /></div>Most peaks remained have a &gt;5 fold enrichment, should be fine as well.&nbsp;</div></div>