---
layout: post
title: methods of calling differential region of ChIP-seq
date: '2013-08-07T22:25:00.001-04:00'
author: Xianjun Dong
tags:
- next-generation sequencing
- ChIPseq
- R
- MACS
modified_time: '2013-08-07T23:07:29.394-04:00'
blogger_id: tag:blogger.com,1999:blog-12445049.post-5101070527196404876
blogger_orig_url: http://onetipperday.blogspot.com/2013/08/methods-of-calling-differential-region.html
---

Related papers to read:<br /><br /><b>Model-based Analysis of ChIP-Seq (MACS)</b><br /><div>MACS can also be applied to differential binding between two conditions by treating one of the samples as the control. Since peaks from either sample are likely to be biologically meaningful in this case, we cannot use a sample swap to calculate FDR, and the data quality of each sample needs to be evaluated against a real control.<br /><blockquote class="tr_bq"><b>Curr Protoc Bioinformatics. Author manuscript; available in PMC 2012 June 1.</b>MACS empirically calculates FDR based on the number of peaks from control over ChIP that&nbsp;are called at the same p-value cutoff. Therefore if no control data is available, the FDR column&nbsp;does not exist in the output tabular file. Technically, MACS can also be applied to identify&nbsp;differential peaks between two conditions by treating one of the samples as the control. However, calculated FDR value should be ignored, as peaks from either sample are likely to&nbsp;be biologically meaningful in this case. &nbsp;</blockquote><blockquote class="tr_bq"><b>http://www.nature.com/nprot/journal/v7/n9/full/nprot.2012.101.html</b>the warning message 'unbalanced reads between treatment and control' means that the FDR of the resulting peaks will be overestimated when the control sample has more reads and will be underestimated when the ChIP-seq sample is sequenced more deeply.</blockquote></div><div><br /></div><b>MAnorm: a robust model for quantitative comparison of ChIP-Seq data sets</b><br /><div>ChIPdiff and MACS identified four to six times more target regions associated with significantly increased ChIP-Seq signals for K562 cells compared with those found for H1 ES cells, whereas MAnorm yielded a similar number of cell type-biased peaks in each cell line. To compare the enrichment of cell type-specifically expressed genes in the sets of target genes of the differential binding regions discovered by the three methods, we selected the same number of target genes associated with top differential binding regions identified by each method. The target genes of top differential binding regions identified by MAnorm contained similar numbers of H1 ES cell highly expressed genes but a greater number of K562 cell highly expressed genes compared to those identified by ChIPdiff and MACS (Supplementary Table 1 in Additional file <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3439967/#S4">4</a>), suggesting MAnorm performs better in detecting differentially binding regions than the other two methods. Importantly, the fold changes of differential binding given by ChIPdiff and MACS were based on the total number of reads, which may not be appropriate, as discussed above. Additionally, MAnorm showed even better enrichment of cell type-specifically expressed genes in differential binding region targets than the method developed by Taslim et al. [<a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3439967/#B12">12</a>] when applied to ChIP-Seq data presented in their study (Supplementary Table 2 in Additional file <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3439967/#S4">4</a>).</div><div><div><span style="background-color: white; font-family: Georgia, 'Times New Roman', serif; font-size: 14.666666984558105px; line-height: 14px;"><br /></span></div><div><table border="2" cellspacing="0"><tbody><tr><th align="center" bgcolor="#B0B0B0" height="64" width="24"><b>1</b></th><td colspan="4"><span style="font-family: Arial; font-size: medium;"><b>Supplementary Table 1</b>. Enrichment of cell-type differentially expressed genes in genes near differential binding peaks defined by MAnorm, MACS and ChIPdiff. To compare the enrichment scores, we selected the same number of target genes associated with top differential binding regions identified by MAnorm with those identified by other two methods.</span></td></tr><tr><th align="center" bgcolor="#B0B0B0" height="53" width="24"><b>2</b></th><td colspan="4"><span style="font-family: Arial; font-size: medium;"><b>Supplementary Table 1A</b>. Enrichment of genes more highly expressed in H1 ES cells (as compared to K562) in genes near H1 ES enriched peaks&nbsp; (as compared to K562) defined by MAnorm, MACS and ChIPdiff</span></td></tr><tr><th align="center" bgcolor="#B0B0B0" height="42" width="24"><b>3</b></th><td width="552"><span style="font-family: Arial; font-size: medium;">H3K27ac H1 ES-enriched target genes</span></td><td align="center" width="174"><span style="font-family: Arial; font-size: medium;">Number of Genes</span></td><td align="center" width="177"><span style="font-family: Arial; font-size: medium;">Overlap with H1 ES up-regulated genes</span></td><td align="center" width="183"><span style="font-family: Arial; font-size: medium;">Enrichment Score</span></td></tr><tr><th align="center" bgcolor="#B0B0B0" height="19" width="24"><b>4</b></th><td bgcolor="#FFFF00" width="552"><span style="font-family: Arial; font-size: medium;">MAnorm (M&gt;1)</span></td><td align="center" bgcolor="#FFFF00" width="174"><span style="font-family: Arial; font-size: medium;">2680</span></td><td align="center" bgcolor="#FFFF00" width="177"><span style="font-family: Arial; font-size: medium;">1243</span></td><td align="center" bgcolor="#FFFF00" width="183"><span style="font-family: Arial; font-size: medium;">2.49</span></td></tr><tr><th align="center" bgcolor="#B0B0B0" height="18" width="24"><b>5</b></th><td width="552"><span style="font-family: Arial; font-size: medium;">ChIPdiff (default)</span></td><td align="center" width="174"><span style="font-family: Arial; font-size: medium;">1467</span></td><td align="center" width="177"><span style="font-family: Arial; font-size: medium;">884</span></td><td align="center" width="183"><span style="font-family: Arial; font-size: medium;">3.24</span></td></tr><tr><th align="center" bgcolor="#B0B0B0" height="45" width="24"><b>6</b></th><td bgcolor="#FFFF00" width="552"><span style="font-family: Arial; font-size: medium;">MAnorm (top 1467 genes; same number of genes as identified by ChIPdiff with default settings)</span></td><td align="center" bgcolor="#FFFF00" width="174"><span style="font-family: Arial; font-size: medium;">1467</span></td><td align="center" bgcolor="#FFFF00" width="177"><span style="font-family: Arial; font-size: medium;">941</span></td><td align="center" bgcolor="#FFFF00" width="183"><span style="font-family: Arial; font-size: medium;">3.45</span></td></tr><tr><th align="center" bgcolor="#B0B0B0" height="18" width="24"><b>7</b></th><td width="552"><span style="font-family: Arial; font-size: medium;">MACS (<i>P</i>&lt;1e-6)</span></td><td align="center" width="174"><span style="font-family: Arial; font-size: medium;">1589</span></td><td align="center" width="177"><span style="font-family: Arial; font-size: medium;">993</span></td><td align="center" width="183"><span style="font-family: Arial; font-size: medium;">3.36</span></td></tr><tr><th align="center" bgcolor="#B0B0B0" height="35" width="24"><b>8</b></th><td bgcolor="#FFFF00" width="552"><span style="font-family: Arial; font-size: medium;">MAnorm (top 1589 genes; same number of genes as identified by MACS with&nbsp;<i>P</i>&lt;1e-6)</span></td><td align="center" bgcolor="#FFFF00" width="174"><span style="font-family: Arial; font-size: medium;">1589</span></td><td align="center" bgcolor="#FFFF00" width="177"><span style="font-family: Arial; font-size: medium;">987</span></td><td align="center" bgcolor="#FFFF00" width="183"><span style="font-family: Arial; font-size: medium;">3.34</span></td></tr><tr><th align="center" bgcolor="#B0B0B0" height="10" width="24"><b>9</b></th><td width="552"></td><td width="174"></td><td width="177"></td><td width="183"></td></tr><tr><th align="center" bgcolor="#B0B0B0" height="9" width="24"><b>10&nbsp;</b></th><td width="552"></td><td width="174"></td><td width="177"></td><td width="183"></td></tr><tr><th align="center" bgcolor="#B0B0B0" height="40" width="24"><b>11&nbsp;</b></th><td colspan="4"><span style="font-family: Arial; font-size: medium;"><b>Supplementary Table 1B</b>. Enrichment of K562 higher expressed genes (as compared to H1 ES) in genes near K562 enriched peaks (as compared to H1 ES) defined by MAnorm, MACS and ChIPdiff</span></td></tr><tr><th align="center" bgcolor="#B0B0B0" height="39" width="24"><b>12</b></th><td width="552"><span style="font-family: Arial; font-size: medium;">H3K27ac K562-enriched target genes</span></td><td align="center" width="174"><span style="font-family: Arial; font-size: medium;">Number of Genes</span></td><td align="center" width="177"><span style="font-family: Arial; font-size: medium;">Overlap with K562 up-regulated genes</span></td><td align="center" width="183"><span style="font-family: Arial; font-size: medium;">Enrichment Score</span></td></tr><tr><th align="center" bgcolor="#B0B0B0" height="19" width="24"><b>13</b></th><td bgcolor="#FFFF00" width="552"><span style="font-family: Arial; font-size: medium;">MAnorm (M&lt;-1)</span></td><td align="center" bgcolor="#FFFF00" width="174"><span style="font-family: Arial; font-size: medium;">2694</span></td><td align="center" bgcolor="#FFFF00" width="177"><span style="font-family: Arial; font-size: medium;">895</span></td><td align="center" bgcolor="#FFFF00" width="183"><span style="font-family: Arial; font-size: medium;">2.78</span></td></tr><tr><th align="center" bgcolor="#B0B0B0" height="18" width="24"><b>14</b></th><td width="552"><span style="font-family: Arial; font-size: medium;">ChIPdiff (default)</span></td><td align="center" width="174"><span style="font-family: Arial; font-size: medium;">6733</span></td><td align="center" width="177"><span style="font-family: Arial; font-size: medium;">1402</span></td><td align="center" width="183"><span style="font-family: Arial; font-size: medium;">1.74</span></td></tr><tr><th align="center" bgcolor="#B0B0B0" height="18" width="24"><b>15</b></th><td width="552"><span style="font-family: Arial; font-size: medium;">ChIPdiff (confidence threshold=0.9999999999)</span></td><td align="center" width="174"><span style="font-family: Arial; font-size: medium;">2291</span></td><td align="center" width="177"><span style="font-family: Arial; font-size: medium;">697</span></td><td align="center" width="183"><span style="font-family: Arial; font-size: medium;">2.55</span></td></tr><tr><th align="center" bgcolor="#B0B0B0" height="62" width="24"><b>16</b></th><td bgcolor="#FFFF00" width="552"><span style="font-family: Arial; font-size: medium;">MAnorm (top 2291 genes; same number of genes as identified by ChIPdiff with confidence threshold=0.9999999999)</span></td><td align="center" bgcolor="#FFFF00" width="174"><span style="font-family: Arial; font-size: medium;">2291</span></td><td align="center" bgcolor="#FFFF00" width="177"><span style="font-family: Arial; font-size: medium;">820</span></td><td align="center" bgcolor="#FFFF00" width="183"><span style="font-family: Arial; font-size: medium;">3.00</span></td></tr><tr><th align="center" bgcolor="#B0B0B0" height="18" width="24"><b>17</b></th><td width="552"><span style="font-family: Arial; font-size: medium;">MACS (P&lt;1e-6)</span></td><td align="center" width="174"><span style="font-family: Arial; font-size: medium;">9346</span></td><td align="center" width="177"><span style="font-family: Arial; font-size: medium;">1600</span></td><td align="center" width="183"><span style="font-family: Arial; font-size: medium;">1.43</span></td></tr><tr><th align="center" bgcolor="#B0B0B0" height="18" width="24"><b>18</b></th><td width="552"><span style="font-family: Arial; font-size: medium;">MACS (P&lt;1e-150)</span></td><td align="center" width="174"><span style="font-family: Arial; font-size: medium;">1556</span></td><td align="center" width="177"><span style="font-family: Arial; font-size: medium;">567</span></td><td align="center" width="183"><span style="font-family: Arial; font-size: medium;">3.05</span></td></tr><tr><th align="center" bgcolor="#B0B0B0" height="48" width="24"><b>19</b></th><td bgcolor="#FFFF00" width="552"><span style="font-family: Arial; font-size: medium;">MAnorm (top 1556 genes; same number of genes as identified by MACS with P&lt;1e-6))</span></td><td align="center" bgcolor="#FFFF00" width="174"><span style="font-family: Arial; font-size: medium;">1556</span></td><td align="center" bgcolor="#FFFF00" width="177"><span style="font-family: Arial; font-size: medium;">644</span></td><td align="center" bgcolor="#FFFF00" width="183"><span style="font-family: Arial; font-size: medium;">3.47</span></td></tr><tr><th align="center" bgcolor="#B0B0B0" width="24"><b>20</b></th><td width="552"><br /></td><td width="174"><br /></td><td width="177"><br /></td><td width="183"><br /></td></tr></tbody></table></div><div><br /><br /><b>diffReps: Detecting Differential Chromatin Modification Sites from ChIP-seq Data with Biological Replicates</b><br />...shows that diffReps is the most sensitive method among all the methods compared, followed by edgeR, DESeq, ChIPDiff and, lastly, CCAT+DESeq. At each cutoff, diffReps (negative binomial test) typically detects a few thousands more differential sites than the secondly ranked method, edgeR.<br />The tool is in Perl (https://code.google.com/p/diffreps/)</div><div><br /><span style="font-size: large;">Related tools in R:</span><br /><br /><b>Detecting differential binding of transcription factors with ChIP-seq</b>, Bioinformatics, 28, 121-122, doi: 10.1093/bioinformatics/btr605<br />DBChIP: http://pages.cs.wisc.edu/~kliang/DBChIP/DBChIP.pdf<br /><br /><b>DiffBind : differential binding analysis of ChIP-Seq peak data</b>http://bioconductor.org/packages/2.12/bioc/vignettes/DiffBind/inst/doc/DiffBind.pdf</div></div><div><br /></div><div>I'd like to make an assessment for the tools later.&nbsp;</div>