---
layout: post
title: R / Bioconductor for High-Throughput Sequence Analysis
date: '2013-12-06T11:18:00.004-05:00'
author: Xianjun Dong
tags:
- next-generation sequencing
- bioC
- R
modified_time: '2013-12-06T11:18:56.504-05:00'
blogger_id: tag:blogger.com,1999:blog-12445049.post-4962899602023054783
blogger_orig_url: http://onetipperday.blogspot.com/2013/12/r-bioconductor-for-high-throughput.html
---

I would like to recommend a recent workshop material on R/Bioconductor from Marc Carlson et al.<br /><br />http://www.bioconductor.org/help/course-materials/2013/SeattleMay2013/<br /><br />PDF: <a href="http://www.bioconductor.org/help/course-materials/2013/SeattleMay2013/IntermediateSequenceAnalysis2013.pdf">IntermediateSequenceAnalysis2013.pdf</a><br />R script:&nbsp;<a href="http://www.bioconductor.org/help/course-materials/2013/SeattleMay2013/IntermediateSequenceAnalysis2013.R">IntermediateSequenceAnalysis2013.R</a><br /><br />Here is the TOC list if you want to read more (highlight for suggested reading list):<br /><br /><a name='more'></a><br /><span style="font-family: Georgia, Times New Roman, serif;">I R / Bioconductor 5</span><br /><span style="font-family: Georgia, Times New Roman, serif;">2 R 6</span><br /><span style="font-family: Georgia, Times New Roman, serif;">2.1 Statistical analysis and comprehension . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 6</span><br /><span style="font-family: Georgia, Times New Roman, serif;">2.2 Basics of R . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 6</span><br /><span style="font-family: Georgia, Times New Roman, serif;">2.2.1 Essential data types . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 7</span><br /><span style="font-family: Georgia, Times New Roman, serif;">2.2.2 S3 (and S4) classes . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 10</span><br /><span style="font-family: Georgia, Times New Roman, serif;">2.2.3 Functions . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 10</span><br /><span style="font-family: Georgia, Times New Roman, serif;">2.3 In and out of trouble . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 16</span><br /><span style="background-color: yellow; font-family: Georgia, Times New Roman, serif;">2.3.1 Warnings, errors, and debugging . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 16</span><br /><span style="background-color: yellow; font-family: Georgia, Times New Roman, serif;">2.3.2 Effi cient R code . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 18</span><br /><span style="font-family: Georgia, Times New Roman, serif;">2.4 Packages . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 21</span><br /><span style="font-family: Georgia, Times New Roman, serif;">2.5 Help! . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 22</span><br /><span style="font-family: Georgia, Times New Roman, serif;">3 Bioconductor 25</span><br /><span style="background-color: yellow; font-family: Georgia, Times New Roman, serif;">3.1 High-throughput sequence analysis . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 26</span><br /><span style="background-color: yellow; font-family: Georgia, Times New Roman, serif;">3.2 Statistical programming . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 26</span><br /><span style="background-color: yellow; font-family: Georgia, Times New Roman, serif;">3.3 Bioconductor packages for high-throughput sequence analysis . . . . . . . . . . . . . . . . . . 28</span><br /><span style="font-family: Georgia, Times New Roman, serif;">3.4 S4 Classes and methods . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 28</span><br /><span style="font-family: Georgia, Times New Roman, serif;">3.5 Help! . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 31</span><br /><span style="font-family: Georgia, Times New Roman, serif;">4 Sequencing 32</span><br /><span style="font-family: Georgia, Times New Roman, serif;">4.1 Technologies . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 32</span><br /><span style="font-family: Georgia, Times New Roman, serif;">4.2 Data . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 33</span><br /><span style="font-family: Georgia, Times New Roman, serif;">4.2.1 A running example: the pasilla data set . . . . . . . . . . . . . . . . . . . . . . . . . . 33</span><br /><span style="font-family: Georgia, Times New Roman, serif;">4.2.2 Work&nbsp;</span><span style="font-family: Georgia, 'Times New Roman', serif;">flows . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 33</span><br /><span style="font-family: Georgia, Times New Roman, serif;">5 Strings and Reads 36</span><br /><span style="font-family: Georgia, Times New Roman, serif;">5.1 DNA (and other) Strings with the Biostrings package . . . . . . . . . . . . . . . . . . . . . . 36</span><br /><span style="font-family: Georgia, Times New Roman, serif;">5.2 Reads and the ShortRead package . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 39</span><br /><span style="font-family: Georgia, Times New Roman, serif;">6 Ranges and Alignments 44</span><br /><span style="font-family: Georgia, Times New Roman, serif;">6.1 Ranges and the GenomicRanges package . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 44</span><br /><span style="font-family: Georgia, Times New Roman, serif;">6.2 Alignments and the Rsamtools Package . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 50</span><br /><span style="background-color: yellow;">7 Strategies for working with large data 54</span><br /><span style="background-color: yellow;">7.1 Restriction . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 54</span><br /><span style="background-color: yellow;">7.2 Sampling . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 56</span><br /><span style="background-color: yellow;">7.3 Iteration . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 57</span><br /><span style="background-color: yellow;">7.4 Parallel evaluation . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 59</span><br /><span style="background-color: yellow;">7.5 And. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 61</span><br />II Diff erential Representation 62<br />8 RNA-seq Work Flows 63<br />8.1 Varieties of RNA-seq . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 63<br />8.2 Work flows and upstream analysis . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 63<br />8.2.1 Experimental design . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 64<br />8.2.2 Wet-lab protocols, sequencing, and alignment . . . . . . . . . . . . . . . . . . . . . . . 64<br />8.3 Statistical analysis . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 65<br />8.3.1 Summarizing . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 65<br />8.3.2 <span style="background-color: yellow;">Normalization</span> . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 66<br />8.3.3 Error model . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 66<br />8.3.4 Multiple comparison . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 66<br />8.3.5 Bioconductor software . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 67<br /><span style="background-color: yellow;">9 Counting RNA-seq reads 68</span><br />9.1 `Phenotypic' data . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 69<br />9.2 Alignments . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 70<br />9.2.1 Alignment, subsetting, and access in R . . . . . . . . . . . . . . . . . . . . . . . . . . . 70<br />9.2.2 Exploration . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 71<br />9.3 Counting . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 73<br />9.4 SummarizedExperiment . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 75<br />9.5 Additional counting and annotation opportunities . . . . . . . . . . . . . . . . . . . . . . . . 77<br />9.5.1 Counting genes, counting transcripts . . . . . . . . . . . . . . . . . . . . . . . . . . . . 77<br />9.5.2 Counting by other means . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 78<br />9.5.3 Using other annotation resources . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 78<br />10 DESeq Work Flow Exercises 79<br />10.1 Data input and preparation . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 79<br />10.2 Inference . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 79<br />10.3 Independent  filtering . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 80<br /><span style="background-color: yellow;">10.4 Data quality assessment . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 80</span><br />10.4.1 Preliminary transformation . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 80<br />10.4.2 Quality assessment . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 80<br />10.5 Frequently asked questions . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 80<br />III Variant Calls 81<br />11 Variant Work Flows 82<br />11.1 Variants . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 82<br />11.1.1 Varieties of variant-related work flows . . . . . . . . . . . . . . . . . . . . . . . . . . . 82<br />11.1.2 Work flows . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 82<br />11.1.3 Bioconductor software . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 82<br />11.2 VariantTools . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 83<br />11.2.1 Example data: lung cancer cell lines . . . . . . . . . . . . . . . . . . . . . . . . . . . . 83<br />11.2.2 Calling single-sample variants . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 83<br />11.2.3 Additional work flows . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 84<br />12 Working with Called Variants 85<br />12.1 Variant call format (VCF) fi les with VariantAnnotation . . . . . . . . . . . . . . . . . . . . . 85<br />12.1.1 Data input . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 85<br />12.2 SNP Annotation . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 87<br />12.3 Annotation with ensemblVEP . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 92<br />12.4 Large-scale fi ltering . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 92<br />IV Annotation and Visualization 95<br />13 Gene-centric Annotation 96<br />13.1 Gene-centric annotations with AnnotationDbi . . . . . . . . . . . . . . . . . . . . . . . . . . . 96<br />13.2 biomaRt and other web-based resources . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 99<br />13.2.1 Using biomaRt . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 100<br />14 Genomic Annotation 101<br />14.1 AnnotationHub . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 101<br />14.2 Whole genome sequences . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 101<br />14.3 Gene models . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 101<br />14.3.1 TxDb.* packages for model organisms . . . . . . . . . . . . . . . . . . . . . . . . . . . 101<br /><span style="background-color: yellow;">14.4 UCSC tracks . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 103</span><br />14.4.1 Easily creating TranscriptDb objects from GTF  files . . . . . . . . . . . . . . . . . . . 103<br /><span style="background-color: yellow;">15 Visualizing Sequence Data 108</span><br />15.1 Gviz . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 108<br />15.2 ggbio . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 109<br />15.3 shiny for easy interactive reports . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 109