---
layout: post
title: The same gene name, but different Ensembl/Gencode ID
date: '2013-08-12T22:51:00.000-04:00'
author: Xianjun Dong
tags:
- gencode
- ensembl
modified_time: '2013-08-12T23:02:34.274-04:00'
blogger_id: tag:blogger.com,1999:blog-12445049.post-2379389720087059981
blogger_orig_url: http://onetipperday.blogspot.com/2013/08/the-same-gene-name-but-different.html
---

I got aware of this when I was working on the latest GENCODE annotation v17. Here is it:<br /><br /><span style="font-family: Courier New, Courier, monospace;">annotationURL=ftp://ftp.sanger.ac.uk/pub/gencode/release_17/gencode.v17.annotation.gtf.gz&nbsp;</span><br /><span style="font-family: Courier New, Courier, monospace;">curl -0 $annotationURL | zcat | fgrep -w gene | sed 's/[";]//g;' |&nbsp;awk '{OFS="\t"; print $1,$2,$4,$5,$7,$10,$14,$16,$18;}' &nbsp;| cut -f9 | sort | uniq -d</span><br /><span style="font-family: Courier New, Courier, monospace;"><b>5S_rRNA</b></span><br /><span style="font-family: Courier New, Courier, monospace;"><b>61E3.4</b></span><br /><span style="font-family: Courier New, Courier, monospace;"><b>A-575C2.4</b></span><br /><span style="font-family: Courier New, Courier, monospace;"><b>AAK1</b></span><br /><span style="font-family: Courier New, Courier, monospace;"><b>ACA59</b></span><br /><span style="font-family: Courier New, Courier, monospace;"><b>ACA64</b></span><br /><span style="font-family: Courier New, Courier, monospace;"><b>ACE</b></span><br /><span style="font-family: Courier New, Courier, monospace;"><b>AJ271736.10</b></span><br /><span style="font-family: Courier New, Courier, monospace;"><b>AK3</b></span><br /><span style="font-family: Courier New, Courier, monospace;"><b>AKAP17A</b></span><br /><span style="font-family: Courier New, Courier, monospace;"><b>AL732314.1</b></span><br /><span style="font-family: Courier New, Courier, monospace;"><b>ALG9</b></span><br /><span style="font-family: Courier New, Courier, monospace;"><b>AMDP1</b></span><br /><span style="font-family: Courier New, Courier, monospace;"><b>APOC4</b></span><br /><span style="font-family: Courier New, Courier, monospace;"><b>AQP1</b></span><br /><span style="font-family: Courier New, Courier, monospace;"><b>ARHGAP19</b></span><br /><span style="font-family: Courier New, Courier, monospace;"><b>...</b></span><br /><span style="font-family: Courier New, Courier, monospace;"><b>...</b></span><br /><span style="font-family: Courier New, Courier, monospace;"><b>...</b></span><br /><span style="font-family: Courier New, Courier, monospace;"><b>ZNF585B</b></span><br /><span style="font-family: Courier New, Courier, monospace;"><b>ZNF607</b></span><br /><span style="font-family: Courier New, Courier, monospace;"><b>ZNF625</b></span><br /><span style="font-family: Courier New, Courier, monospace;"><b>ZNF668</b></span><br /><span style="font-family: Courier New, Courier, monospace;"><b>ZNF709</b></span><br /><span style="font-family: Courier New, Courier, monospace;"><b>ZNF747</b></span><br /><span style="font-family: Courier New, Courier, monospace;"><b>ZNF763</b></span><br /><span style="font-family: Courier New, Courier, monospace;"><b>ZNF788</b></span><br /><br />In total, there are 400 genes that have &gt;1 different&nbsp;GENCODE&nbsp;ID. For example,<br /><br /><span style="font-family: Courier New, Courier, monospace;">curl -0 $annotationURL | zcat&nbsp;| fgrep -w gene | sed 's/[";]//g;' | awk '{OFS="\t"; print $1,$2,$4,$5,$7,$10,$14,$16,$18;}' | &nbsp;grep ZNF788</span><br /><span style="font-family: Courier New, Courier, monospace;"><b>chr19<span class="Apple-tab-span" style="white-space: pre;"> </span>HAVANA<span class="Apple-tab-span" style="white-space: pre;"> </span>12203078<span class="Apple-tab-span" style="white-space: pre;"> </span>12248050<span class="Apple-tab-span" style="white-space: pre;"> </span>+<span class="Apple-tab-span" style="white-space: pre;"> </span>ENSG00000214189.3<span class="Apple-tab-span" style="white-space: pre;"> </span>protein_coding<span class="Apple-tab-span" style="white-space: pre;"> </span>KNOWN<span class="Apple-tab-span" style="white-space: pre;"> </span>ZNF788</b></span><br /><span style="font-family: Courier New, Courier, monospace;"><b>chr19<span class="Apple-tab-span" style="white-space: pre;"> </span>ENSEMBL<span class="Apple-tab-span" style="white-space: pre;"> </span>12203078<span class="Apple-tab-span" style="white-space: pre;"> </span>12225491<span class="Apple-tab-span" style="white-space: pre;"> </span>+<span class="Apple-tab-span" style="white-space: pre;"> </span>ENSG00000188474.6<span class="Apple-tab-span" style="white-space: pre;"> </span>protein_coding<span class="Apple-tab-span" style="white-space: pre;"> </span>KNOWN<span class="Apple-tab-span" style="white-space: pre;"> </span>ZNF788</b></span><br /><br />This case is annotated differently in different sources (HAVANA vs. ENSEMBL). But even in the same source, there are cases that are annotated differently, for example:<br /><br /><span style="font-family: Courier New, Courier, monospace;">zcat $annotation | fgrep -w gene | sed 's/[";]//g;' | awk '{OFS="\t"; print $1,$2,$4,$5,$7,$10,$14,$16,$18;}' | fgrep -w ENSEMBL | grep GRIA1</span><br /><span style="font-family: Courier New, Courier, monospace;"><b>chr5<span class="Apple-tab-span" style="white-space: pre;"> </span>ENSEMBL<span class="Apple-tab-span" style="white-space: pre;"> </span>152871759<span class="Apple-tab-span" style="white-space: pre;"> </span>153190785<span class="Apple-tab-span" style="white-space: pre;"> </span>+<span class="Apple-tab-span" style="white-space: pre;"> </span>ENSG00000270065.1<span class="Apple-tab-span" style="white-space: pre;"> </span>protein_coding<span class="Apple-tab-span" style="white-space: pre;"> </span>KNOWN<span class="Apple-tab-span" style="white-space: pre;"> </span>GRIA1</b></span><br /><span style="font-family: Courier New, Courier, monospace;"><b>chr5<span class="Apple-tab-span" style="white-space: pre;"> </span>ENSEMBL<span class="Apple-tab-span" style="white-space: pre;"> </span>152873613<span class="Apple-tab-span" style="white-space: pre;"> </span>153190785<span class="Apple-tab-span" style="white-space: pre;"> </span>+<span class="Apple-tab-span" style="white-space: pre;"> </span>ENSG00000269977.1<span class="Apple-tab-span" style="white-space: pre;"> </span>protein_coding<span class="Apple-tab-span" style="white-space: pre;"> </span>KNOWN<span class="Apple-tab-span" style="white-space: pre;"> </span>GRIA1</b></span><br /><br />Obviously cases like this should be merged into one&nbsp;Gencode&nbsp;ID.<br /><br />Some are even on different chromosomes:<br /><br /><span style="font-family: Courier New, Courier, monospace;">zcat $annotation | fgrep -w gene | sed 's/[";]//g;' | awk '{OFS="\t"; print $1,$2,$4,$5,$7,$10,$14,$16,$18;}' | fgrep -w ENSEMBL | grep ACA59</span><br /><span style="font-family: Courier New, Courier, monospace;"><b>chr2<span class="Apple-tab-span" style="white-space: pre;"> </span>ENSEMBL<span class="Apple-tab-span" style="white-space: pre;"> </span>64110383<span class="Apple-tab-span" style="white-space: pre;"> </span>64110525<span class="Apple-tab-span" style="white-space: pre;"> </span>-<span class="Apple-tab-span" style="white-space: pre;"> </span>ENSG00000251775.1<span class="Apple-tab-span" style="white-space: pre;"> </span>snoRNA<span class="Apple-tab-span" style="white-space: pre;"> </span>NOVEL<span class="Apple-tab-span" style="white-space: pre;"> </span>ACA59</b></span><br /><span style="font-family: Courier New, Courier, monospace;"><b>chr2<span class="Apple-tab-span" style="white-space: pre;"> </span>ENSEMBL<span class="Apple-tab-span" style="white-space: pre;"> </span>179887984<span class="Apple-tab-span" style="white-space: pre;"> </span>179888105<span class="Apple-tab-span" style="white-space: pre;"> </span>-<span class="Apple-tab-span" style="white-space: pre;"> </span>ENSG00000252000.1<span class="Apple-tab-span" style="white-space: pre;"> </span>snoRNA<span class="Apple-tab-span" style="white-space: pre;"> </span>NOVEL<span class="Apple-tab-span" style="white-space: pre;"> </span>ACA59</b></span><br /><span style="font-family: Courier New, Courier, monospace;"><b>chr11<span class="Apple-tab-span" style="white-space: pre;"> </span>ENSEMBL<span class="Apple-tab-span" style="white-space: pre;"> </span>114998938<span class="Apple-tab-span" style="white-space: pre;"> </span>114999093<span class="Apple-tab-span" style="white-space: pre;"> </span>-<span class="Apple-tab-span" style="white-space: pre;"> </span>ENSG00000252870.1<span class="Apple-tab-span" style="white-space: pre;"> </span>snoRNA<span class="Apple-tab-span" style="white-space: pre;"> </span>NOVEL<span class="Apple-tab-span" style="white-space: pre;"> </span>ACA59</b></span><br /><br />How come different genes/loci are assigned a same gene name?<br /><br />Looks confusing! Hope someone from Gencode or Ensembl can help to understand this.