---
layout: post
title: compatible command line for different input format (*.fastq, *.fq, or *.fq.gz)
date: '2013-08-02T16:39:00.001-04:00'
author: Xianjun Dong
tags:
- onelinecommand
- next-generation sequencing
- linux
- bowtie
modified_time: '2013-08-02T16:39:32.997-04:00'
blogger_id: tag:blogger.com,1999:blog-12445049.post-6087118246510313456
blogger_orig_url: http://onetipperday.blogspot.com/2013/08/compatible-command-line-for-different.html
---

For example, if I want to run bowtie with fastq or zipped fastq files, here is the workout:<br /><br /><span style="font-family: Courier New, Courier, monospace;">(([[ $i =~ "fq|fastq$" ]] &amp;&amp; cat $i) || ([[ $i =~ "gz$" ]] &amp;&amp; zcat $i)) | bowtie genome -v 1 -M 1 --best --strata -q - -S accepted_hits.sam 2&gt; mapping.summary</span><br /><br />In above example, bowtie will take input in format of *.fastq, *.fq, or *.gz and output the best hit (or one of the best hits if multiple hits map equally best) for each read with 1 mismatch allowed, and redirect the log to mapping.summary file.<br /><div><br /></div>