---
layout: post
title: Cufflinks mask option (-M/--mask-file) works when ...
date: '2014-10-08T17:06:00.000-04:00'
author: Xianjun Dong
tags:
- cufflinks
- next-generation sequencing
- RNAseq
modified_time: '2015-05-08T12:03:25.804-04:00'
blogger_id: tag:blogger.com,1999:blog-12445049.post-5146672294829222563
blogger_orig_url: http://onetipperday.blogspot.com/2014/10/cufflinks-mask-option-m-mask-file-works.html
---

Obviously I am not the only one who had questions on the "-M/--mask-file" mask GTF option in Cufflinks:<br /><br /><a href="http://seqanswers.com/forums/showthread.php?t=8190">http://seqanswers.com/forums/showthread.php?t=8190</a><br /><a href="https://www.biostars.org/p/110289/">https://www.biostars.org/p/110289/</a><br /><a href="http://seqanswers.com/forums/showthread.php?t=29975">http://seqanswers.com/forums/showthread.php?t=29975</a><br /><div style="background-color: white; border: 0px; color: #222222; font-family: Arial, Helvetica, sans-serif; font-size: 13px; margin: 0px; padding: 0px; vertical-align: baseline;"><br /></div>And too bad that no one from the Texedo group ever threw a piece of clue!<br /><br />Here are few tips I found necessary to share in order to have it work:<br /><div style="border: 0px; margin: 0px; padding: 0px; vertical-align: baseline;"><div style="background-color: white; color: #222222; font-family: Arial, Helvetica, sans-serif; font-size: 13px;"><br /></div><span style="color: #222222; font-family: Arial, Helvetica, sans-serif; font-size: x-small;"><span style="background-color: white;">1.&nbsp;</span></span>The mask GTF file should have all 9 fields in <a href="http://genome.ucsc.edu/FAQ/FAQformat.html#format3">required format</a>. For example, the strand column should be '+', '-', or '.', not anything else. GTF/GFF file can be extracted from GENCODE (http://www.gencodegenes.org/) or <a href="http://onetipperday.blogspot.com/2012/08/how-to-get-trnarrnamitochondrial-gene.html?utm_source=BP_recent">downloaded from UCSC Table browser</a>. It can be also converted from a bed file <a href="http://onetipperday.blogspot.com/2012/08/convert-bed-to-gtf.html">using Kent's&nbsp;<b style="background-color: #fefdfa; color: #333333; font-family: 'Courier New', Courier, monospace; font-size: 13px; line-height: 18.200000762939453px;">bedToGenePred --&gt;&nbsp;</b><b style="background-color: #fefdfa; color: #333333; font-family: 'Courier New', Courier, monospace; font-size: 13px; line-height: 18.200000762939453px;">genePredToGtf</b></a>. But be aware that that the bed file should have at least 6 columns (i.e. including strand column), otherwise the converted GTF file will have a "^@" in the strand column, which results in an invalid GTF.<br /><br />For example, if you want to exclude all reads mapped to human mitochondrial genome, &nbsp;you can use<br /><span style="font-family: Courier New, Courier, monospace;">echo "chrM 0 16571 mt 0 ." | bedToGenePred stdin stdout | genePredToGtf file stdin chrM.gtf</span><br /><br />2. "-M" option also works for de novo assembly (<span style="font-family: Courier New, Courier, monospace;">cufflinks -g</span>).<br /><br />3. Using "-M" option should theoretically increase the FPKM value (comparing to no mask). So, if you observed opposite tread, there must be something wrong.<br /><br />4. If you expect a lot of reads from the mask regions (e.g. chrM, rRNAs), you can substract the masked reads from your bam file before feeding to cufflinks, for example using "<span style="font-family: Courier New, Courier, monospace;">samtools view -L retained_region.bed</span>". </div>