---
layout: post
title: 'Memory issue for large alignment file when doing de novo assembly '
date: '2014-10-10T15:43:00.000-04:00'
author: Xianjun Dong
tags:
- high-performance computing
- NGS
- cufflinks
- next-generation sequencing
modified_time: '2014-10-10T15:43:03.367-04:00'
blogger_id: tag:blogger.com,1999:blog-12445049.post-2380843574571428913
blogger_orig_url: http://onetipperday.blogspot.com/2014/10/memory-issue-for-large-alignment-file.html
---

Memory becomes a bottleneck when the sequencing file is bigger and bigger nowadays. This is specially an issue for de novo transcriptome assembly using RNA-seq data from species like human. For Trinity, "<a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3875132/">a typical configuration is a multi-core server with 256 GB to 1 TB of RAM</a>". "<a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3875132/">Trinity partitions RNA-Seq data into many independent de Bruijn graphs, ideally one graph per expressed gene, and uses parallel computing to reconstruct transcripts from these graphs, including alternatively spliced isoforms</a>." If Trinity contructs one de Bruijn graph per gene, I don't know why it still needs such a large memory. For Cufflinks (-g option), I already can see it consumes 40G memory for 1/10 of chr1 (given a 4G bam file). Cufflinks constructs a DAG in memory for the given alignment. Hopefully it's one DAG per chromosome, not a DAG for the whole genome. But even though, chr1 has 249 billion base pairs, it still requires a lot of memory... So sad!<br /><br />Of course, if you have a machine with super large memory, this won't be a problem. But this is not the case usually. For me, most of our nodes have maximally 90G memory. <br /><br />So, how to have it run?<br /><br />Here is few tips I can think of:<br /><br />1. split the alignment into different chromosome, e.g. chr1.bam, chr2,bam etc. and then call de novo on each of files. <br /><br />2. down-sampling the bam file if it has high coverage. Here is a post about how to down-sample your bam file (https://www.biostars.org/p/4332/). Basically, you can use "<span style="font-family: Courier New, Courier, monospace;">samtools view -s</span>" or "<span style="font-family: Courier New, Courier, monospace;">sambamba view -s</span>", or GATK's&nbsp;<span style="font-family: Courier New, Courier, monospace;">randomSampleFromStream.pl</span> for downsampling. <br /><br />3. You may also want to remove the PCR artifacts, by "<span style="font-family: Courier New, Courier, monospace;">samtools rmdup</span>" or "<span style="font-family: 'Courier New', Courier, monospace;">sambamba&nbsp;</span><span style="font-family: Courier New, Courier, monospace;">markdup</span>".