---
layout: post
title: Note from GEUVADIS paper
date: '2015-07-28T15:41:00.002-04:00'
author: Xianjun Dong
tags:
- normalization
- eQTL
- next-generation sequencing
- RNAseq
modified_time: '2015-07-28T15:41:46.617-04:00'
thumbnail: http://3.bp.blogspot.com/-UMd-Dg_7RZc/VGETa_mC13I/AAAAAAAAAvQ/78-RpaMYEkE/s72-c/nbt.2702-F2.jpg
blogger_id: tag:blogger.com,1999:blog-12445049.post-7973750307537308911
blogger_orig_url: http://onetipperday.blogspot.com/2015/07/note-from-geuvadis-paper.html
---

Note from GEUVADIS papers:<br /><br /><div style="background-color: #efefef; font-family: Arial, Helvetica, sans-serif; font-size: 12px; margin: 0px; padding: 0px;"><span style="text-align: justify;">Lappalainen et al.&nbsp;</span>Nature 2013&nbsp;<span style="text-align: justify;">: Transcriptome and genome sequencing uncovers functional variation in humans,&nbsp;</span><a href="http://dx.doi.org/10.1038/nature12531" style="color: #336699; text-align: justify;">http://dx.doi.org/10.1038/nature12531</a></div><div style="background-color: #efefef; font-family: Arial, Helvetica, sans-serif; font-size: 12px; margin: 0px; padding: 0px;"><b>&nbsp;</b></div><div style="background-color: #efefef; font-family: Arial, Helvetica, sans-serif; font-size: 12px; margin: 0px; padding: 0px;">â€˜t Hoen et al. Nature Biotechnology 2013:<span style="text-align: justify;">&nbsp;Reproducibility of high-throughput mRNA and small RNA sequencing across laboratories,&nbsp;</span><a href="http://dx.doi.org/10.1038/nbt.2702" style="color: #336699; text-align: justify;">http://dx.doi.org/10.1038/nbt.2702</a></div><br /><b>1. how to detect sample outlier?</b><br /><blockquote class="tr_bq">a. before alignment: distance of k-mer profile<br />b. after alignment: Spearman rank correlation between samples --&gt; D-statistics (i.e. &nbsp;the median correlation of one sample against all the other samples)<br />c. gender mismatch: XIST vs. chrY<br />d. ASE bias rate among heterozygous sites</blockquote><br /><div class="separator" style="clear: both; text-align: center;"><a href="https://3.bp.blogspot.com/-UMd-Dg_7RZc/VGETa_mC13I/AAAAAAAAAvQ/78-RpaMYEkE/s1600/nbt.2702-F2.jpg" imageanchor="1" style="margin-left: 1em; margin-right: 1em;"><img border="0" height="361" src="http://3.bp.blogspot.com/-UMd-Dg_7RZc/VGETa_mC13I/AAAAAAAAAvQ/78-RpaMYEkE/s400/nbt.2702-F2.jpg" width="400" /></a></div><br /><br /><b>2. eQTL</b><br /><blockquote class="tr_bq">a. exon/gene quantification<br />b. filter out lowly expressed ones (e.g. 0 in &gt;50% samples)<br />c. for each group, normalize with PEER, adding mean<br />&nbsp; &nbsp; c1. use subset (??e.g. chr20, or chr20- 22??) using K=????0,1,,3,,57,10,13,15,20 for each dataset<br />&nbsp; &nbsp; c2. run eQTL and number of genes (eGenes) for each K.<br />&nbsp; &nbsp; c3. get the optimal K = K(with most number of eQTL genes)<br />&nbsp; &nbsp; c4. run PEER on 20,000 exons to get covairantes for the final normalization<br />&nbsp; &nbsp; c5. final PEER normalization using all dataset, residual + mean as final quantification<br />d. transform the final quantification to standard normal distribution (by ?)<br />e. eQTL using Matrix-eQTL: linear regression of quantification ~ genotypes + genotype_covariates</blockquote><b>3. Differential expression analysis</b><br /><blockquote class="tr_bq">a. TMM normalization (from edgeR)<br />b. filter: genes with more than 5 counts per million in at least 1 sample were analyzed in pairwise population comparisons<br />c. tweeDEseq (good for large samples), significance: FDR &lt; 0.05 and log2 fold change greater than 2</blockquote>