---
layout: post
title: PLINK2 vs. SNAP for linkage disequilibrium (LD) calculation
date: '2015-09-11T10:36:00.003-04:00'
author: Xianjun Dong
tags:
- biology
- LD
- SNAP
- bioinformatics
- GWAS
- PLINK2
- linkage disequilibrium
- SNP
modified_time: '2015-09-11T11:50:49.908-04:00'
blogger_id: tag:blogger.com,1999:blog-12445049.post-369856792850300536
blogger_orig_url: http://onetipperday.blogspot.com/2015/09/linkage-disequilibrium-ld-calculation.html
---

Among other ways to calculate SNPs in linkage disequilibrium (LD), two methods have been used in many literatures: PLINK2 and SNAP<br /><br />With PLINK2, SNPs in LD can be calculated using parameters like:<br /><span style="font-family: Courier New, Courier, monospace;">PLINK2 -r2 dprime --ld-window-kb 1000 --ld-window 10 --ld-window-r2 0.8</span><br />SNAP is a web-based service with pre-calculated data: https://www.broadinstitute.org/mpg/snap. You can set the population, distance limit, r2 threshold.<br /><br />Note that PLINK2 has one more control: <span style="font-family: Courier New, Courier, monospace;">--ld-window</span><br /><br />In its manual (https://www.cog-genomics.org/plink2/ld), it says "By default, when a limited window report is requested, every pair of variants with at least (10-1) variants between them, or more than 1000 kilobases apart". <br /><br />That means <span style="background-color: yellow;">PLINK2 limits SNPs with two kinds of radius: both the genomic distance (<span style="font-family: Courier New, Courier, monospace;">--ld-window-kb</span>) and the number of SNPs apart (<span style="font-family: Courier New, Courier, monospace;">--ld-window</span>), while SNAP only has the distance&nbsp;limit</span>. Both of them&nbsp;have the r2&nbsp;threshold.